{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# UC Severity Qiime Analyses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By RM\n",
    "\n",
    "Environment:Qiime2(2018.4) & Qiime1\n",
    "\n",
    "Project Abstract:\n",
    "\n",
    "Ulcerative colitis (UC) is a chronic autoimmune condition defined by intestinal inflammation and concurrent microbiome dysbiosis. Here, we advance the understanding of host-microbiome interactions governing UC by collecting six metaâ€“omic datasets profiling host and microbial molecules in 40 patients displaying a wide range of clinically assessed disease activity (remission to severe). The six datasets provided unique evidence toward a central hypothesis of proteolysis co-occurring with increased disease activity. Metaproteomics identified Bacteroides proteases as a distinguishing feature of severity. Shotgun metagenomics guided taxonomic inferences and revealed that the Bacteroides association was driven primarily by changes in protein and not DNA abundances. Potential evidence of a host response to Bacteroides serine proteases was found in the increase of serum and fecal serine protease inhibitors. Metapeptidomics added evidence of protease activity as an increase of peptide fragments was present among the patients with high severity. In addition, we compare prediction of severity and clinical parameters between data types. In total, our meta-omic platform has provided compelling integrated evidence for host-microbiome interactions during UC and opens the door for protease inhibition as a therapeutic approach for severe UC patients.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16S.biom\r\n",
      "16S_biom.qza\r\n",
      "16S_biom.qzv\r\n",
      "16S_pmayo_PCoA_Unweighted_CalprotectinSize.png\r\n",
      "16S_pmayo_PCoA_Unweighted_CalprotectinSize.svg\r\n",
      "\u001b[34m1Search_only\u001b[m\u001b[m/\r\n",
      "1search_biom.qza\r\n",
      "1search_biom.qzv\r\n",
      "1search_common_biom.qza\r\n",
      "\u001b[34m2Search\u001b[m\u001b[m/\r\n",
      "2search_biom.qza\r\n",
      "2search_biom.qzv\r\n",
      "2search_common_biom.qza\r\n",
      "2searchpDB_pmayo_PCoA_braycurtis_CalprotectinSize.png\r\n",
      "2searchpDB_pmayo_PCoA_braycurtis_CalprotectinSize.svg\r\n",
      "Bacteroides_Proteases_by_Species_workup.xlsx\r\n",
      "Bacteroides_of_interest.xlsx\r\n",
      "Bacteroides_workup.xlsx\r\n",
      "COG_Spearman_Corrs_Stats_pergene.csv\r\n",
      "Descriptive_Statistics.xlsx\r\n",
      "IGCNormalizedCommonReps.biom\r\n",
      "IGCNormalizedDataAll.biom\r\n",
      "IGC_biom.qza\r\n",
      "IGC_biom.qzv\r\n",
      "IGC_common_biom.qza\r\n",
      "List_of_proteins_NotInSearch1.xlsx\r\n",
      "LowMayoStoolFreq_BH.csv\r\n",
      "Low_severity_remission.csv\r\n",
      "Low_severity_remission_volcanos.ai\r\n",
      "Low_severity_remission_volcanos.png\r\n",
      "Lysozyme_correlations_ordered_MP_corrnonan.csv\r\n",
      "Lysozyme_correlations_ordered_MP_corrwnan.csv\r\n",
      "MG_Bacteroides_Composition.csv\r\n",
      "MG_Enzymes.csv\r\n",
      "MGw0s_LR05_pmayo_Bacteroides.csv\r\n",
      "MPwmin_LR05_pmayo_Bacteroides.csv\r\n",
      "Mayo0Compare.csv\r\n",
      "Mayo0Compare_BH.csv\r\n",
      "Mayo0RemissionCompare.csv\r\n",
      "Mayo0RemissionCompare_BH.csv\r\n",
      "Mayo0_Remission_Volcano.pdf\r\n",
      "Mayo0_Remission_Volcano_TonB.pdf\r\n",
      "Mayo0_Remission_Volcano_TonB.png\r\n",
      "Mayo0_Remission_Volcano_TonB_pi1.pdf\r\n",
      "Mayo0_Remission_Volcano_TonB_pi1.png\r\n",
      "Mayo0_Remission_Volcano_TonB_pi1_BH15.pdf\r\n",
      "Mayo0_Remission_Volcano_expanded.pdf\r\n",
      "Mayo0_Volcano_TonB_pi1.pdf\r\n",
      "Metabolomics.biom\r\n",
      "Metabolomics_biom.qza\r\n",
      "Metabolomics_biom.qzv\r\n",
      "Partial_Mayo_correlations_ordered_MG.csv\r\n",
      "Partial_Mayo_correlations_ordered_MP.csv\r\n",
      "Partial_Mayo_correlations_ordered_MP_corrwnan.csv\r\n",
      "Partial_Mayo_correlations_ordered_nohuman.csv\r\n",
      "\u001b[34mQiime\u001b[m\u001b[m/\r\n",
      "Salmon_CPMs.biom\r\n",
      "Salmon_CPMs_biom.qza\r\n",
      "Serum_Common.biom\r\n",
      "Serum_Common_biom.qza\r\n",
      "Serum_Common_biom.qzv\r\n",
      "Significant_Enzymes_Membranes_Genera.pzfx\r\n",
      "UC_MG_LR05_pmayo_KEGG_ratios.csv\r\n",
      "UC_MG_LR05_pmayo_KEGG_ratios_0s.csv\r\n",
      "UC_MG_LR05_pmayo_KEGG_ratios_nans.csv\r\n",
      "UC_MG_LR05_pmayo_eggNOG_ratios.csv\r\n",
      "UC_MG_LR05_pmayo_eggNOG_ratios_nans.csv\r\n",
      "UC_MP_LR05_Lysozyme_KEGG_ratios_nans.csv\r\n",
      "UC_MP_LR05_pmayo_KEGG_ratios.csv\r\n",
      "UC_MP_LR05_pmayo_KEGG_ratios_nans.csv\r\n",
      "UC_MP_LR05_pmayo_eggNOG_ratios.csv\r\n",
      "UC_MP_LR05_pmayo_eggNOG_ratios_nans.csv\r\n",
      "UC_MP_LR05wmin_pmayo_KEGG_ratios_Bacteroides.csv\r\n",
      "UC_MP_LR05wmin_pmayo_KEGG_ratios_Bacteroides_nostrsplit.csv\r\n",
      "UC_MP_LR05wmin_pmayo_KEGG_ratios_Faecalibacterium.csv\r\n",
      "UC_MP_LR05wmin_pmayo_KEGG_ratios_Faecalibacterium_nostrsplit.csv\r\n",
      "UC_MP_LR05wmin_pmayo_eggNOG_ratios.csv\r\n",
      "UC_MP_LR05wmin_pmayo_eggNOG_ratios_Bacteroides.csv\r\n",
      "UC_MP_LR05wmin_pmayo_eggNOG_ratios_Faecalibacterium.csv\r\n",
      "\u001b[31mUC_Severity_Qiime2_Qiime1.ipynb\u001b[m\u001b[m*\r\n",
      "\u001b[31mUC_Severity_pDB_Qiime2_pseudo1.ipynb\u001b[m\u001b[m*\r\n",
      "\u001b[34mcore-metrics-results_2search\u001b[m\u001b[m/\r\n",
      "\u001b[34mcore-metrics-results_Metabolomics\u001b[m\u001b[m/\r\n",
      "\u001b[34mcore-metrics-results_pDB_noHuman\u001b[m\u001b[m/\r\n",
      "fdr_tsbh.csv\r\n",
      "pDB1_fwdnrvrse_hits.csv\r\n",
      "pDB1_fwdnrvrse_hits.txt\r\n",
      "pDB_Common_nohuman_biom.qza\r\n",
      "pDB_Common_nohuman_biom.qzv\r\n",
      "pDB_Enzymes.csv\r\n",
      "pDB_MP_annotated(nodups).csv\r\n",
      "pDB_MP_annotated.csv\r\n",
      "pMayo2RemissionCompare.csv\r\n",
      "pMayo2_Remission_Volcano.pdf\r\n",
      "pmayo05_MG0MPwmin_KEGG_ratios_(wPseudo)_Workup.xlsx\r\n",
      "pmayo05_MG0sMPwmin_Genus_ratios_(wPseudo)_Workup.xlsx\r\n",
      "pmayo05_MG_Genus_ratios(noPseudo).csv\r\n",
      "pmayo05_MG_Genus_ratios(wPseudo).csv\r\n",
      "pmayo05_MG_Genus_ratios(wPseudo)_workup.csv\r\n",
      "pmayo05_MG_KEGG_ratios(noPseudo).csv\r\n",
      "pmayo05_MG_KEGG_ratios(wPseudo).csv\r\n",
      "pmayo05_MG_eggNOG_ratios_0s(wPseudo).csv\r\n",
      "pmayo05_MP_Genus_ratios(noPseudo).csv\r\n",
      "pmayo05_MP_Genus_ratios(wPseudo).csv\r\n",
      "pmayo05_MPwmin_Genus_ratios(noPseudo).csv\r\n",
      "pmayo05_MPwmin_Genus_ratios_(wPseudo).csv\r\n",
      "pmayo05_MPwmin_KEGG_ratios(noPseudo).csv\r\n",
      "pmayo05_MPwmin_KEGG_ratios_(wPseudo).csv\r\n",
      "pmayo05_MPwmin_eggNOG_ratios_(wPseudo).csv\r\n",
      "pmayo05_pDB_Bacteroides_ratios(noPseudo).csv\r\n",
      "pmayo05_pDB_Bacteroides_ratios(wPseudo).csv\r\n",
      "pmayo_correlations_MG_corrw0s_10ormoresamples.csv\r\n",
      "pmayo_correlations_ordered_MG_corrw0s.csv\r\n",
      "pmayo_correlations_ordered_MG_corrwnan.csv\r\n",
      "pmayo_correlations_ordered_MG_corrwnan10.csv\r\n",
      "pmayo_correlations_ordered_MP_corrwmins.csv\r\n",
      "pmayo_correlations_ordered_MP_corrwnan.csv\r\n",
      "pmayo_correlations_ordered_MP_corrwnan10.csv\r\n",
      "pmayo_correlations_ordered_MP_corrwnan2.csv\r\n",
      "\u001b[34mvmf-map\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "# Initializes the notebook with inline display\n",
    "%matplotlib inline\n",
    "\n",
    "from os import mkdir\n",
    "import os\n",
    "import copy\n",
    "from os.path import abspath, join as pjoin, exists\n",
    "from shutil import copy2, move\n",
    "from time import strftime, strptime\n",
    "from numpy import nan, isnan, arange\n",
    "from pandas import read_csv, Series, DataFrame\n",
    "from IPython.display import Image\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make subset of no-human metaproteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"./2Search/CSVs/NormalizedCommonReps.txt\", \n",
    "                 sep = '\\t', index_col= \"datarest$ProteinID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H16</th>\n",
       "      <th>H3</th>\n",
       "      <th>L4</th>\n",
       "      <th>L11</th>\n",
       "      <th>H8</th>\n",
       "      <th>L3</th>\n",
       "      <th>H11</th>\n",
       "      <th>H20</th>\n",
       "      <th>L10</th>\n",
       "      <th>L7</th>\n",
       "      <th>...</th>\n",
       "      <th>L16</th>\n",
       "      <th>H14</th>\n",
       "      <th>H1</th>\n",
       "      <th>L1</th>\n",
       "      <th>H6</th>\n",
       "      <th>L14</th>\n",
       "      <th>H2</th>\n",
       "      <th>H10</th>\n",
       "      <th>L17</th>\n",
       "      <th>L26</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>datarest$ProteinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>k99_1000928_1</th>\n",
       "      <td>111.069493</td>\n",
       "      <td>94.231149</td>\n",
       "      <td>166.363367</td>\n",
       "      <td>99.088192</td>\n",
       "      <td>54.591755</td>\n",
       "      <td>173.691705</td>\n",
       "      <td>50.662214</td>\n",
       "      <td>134.574253</td>\n",
       "      <td>26.011840</td>\n",
       "      <td>59.754656</td>\n",
       "      <td>...</td>\n",
       "      <td>68.093307</td>\n",
       "      <td>71.056933</td>\n",
       "      <td>33.271179</td>\n",
       "      <td>161.770360</td>\n",
       "      <td>70.540445</td>\n",
       "      <td>101.274311</td>\n",
       "      <td>149.492346</td>\n",
       "      <td>99.904085</td>\n",
       "      <td>58.012858</td>\n",
       "      <td>60.128184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1001938_3</th>\n",
       "      <td>46.630938</td>\n",
       "      <td>320.607204</td>\n",
       "      <td>264.045104</td>\n",
       "      <td>74.501569</td>\n",
       "      <td>33.129914</td>\n",
       "      <td>149.089471</td>\n",
       "      <td>239.364036</td>\n",
       "      <td>60.129463</td>\n",
       "      <td>6.352032</td>\n",
       "      <td>16.593794</td>\n",
       "      <td>...</td>\n",
       "      <td>88.449492</td>\n",
       "      <td>410.993816</td>\n",
       "      <td>15.131730</td>\n",
       "      <td>19.084435</td>\n",
       "      <td>3.638069</td>\n",
       "      <td>2.979719</td>\n",
       "      <td>106.715396</td>\n",
       "      <td>22.169808</td>\n",
       "      <td>66.089185</td>\n",
       "      <td>35.278634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1001990_2</th>\n",
       "      <td>57.391732</td>\n",
       "      <td>47.461439</td>\n",
       "      <td>123.003403</td>\n",
       "      <td>32.235909</td>\n",
       "      <td>66.112950</td>\n",
       "      <td>69.628543</td>\n",
       "      <td>106.518997</td>\n",
       "      <td>705.535557</td>\n",
       "      <td>116.762120</td>\n",
       "      <td>146.875607</td>\n",
       "      <td>...</td>\n",
       "      <td>17.494716</td>\n",
       "      <td>12.224755</td>\n",
       "      <td>34.637102</td>\n",
       "      <td>12.661787</td>\n",
       "      <td>126.771758</td>\n",
       "      <td>104.163969</td>\n",
       "      <td>37.907910</td>\n",
       "      <td>168.860490</td>\n",
       "      <td>30.236346</td>\n",
       "      <td>143.500305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1002150_3</th>\n",
       "      <td>30.935689</td>\n",
       "      <td>125.472858</td>\n",
       "      <td>347.763489</td>\n",
       "      <td>148.560410</td>\n",
       "      <td>39.245119</td>\n",
       "      <td>43.271356</td>\n",
       "      <td>129.515300</td>\n",
       "      <td>79.409804</td>\n",
       "      <td>25.220030</td>\n",
       "      <td>71.962896</td>\n",
       "      <td>...</td>\n",
       "      <td>62.935518</td>\n",
       "      <td>21.794327</td>\n",
       "      <td>36.540982</td>\n",
       "      <td>79.051493</td>\n",
       "      <td>42.634040</td>\n",
       "      <td>61.234510</td>\n",
       "      <td>69.066572</td>\n",
       "      <td>725.394872</td>\n",
       "      <td>136.957015</td>\n",
       "      <td>48.171145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1002370_6</th>\n",
       "      <td>46.583836</td>\n",
       "      <td>164.440568</td>\n",
       "      <td>166.747069</td>\n",
       "      <td>134.089458</td>\n",
       "      <td>14.935078</td>\n",
       "      <td>52.832114</td>\n",
       "      <td>52.968206</td>\n",
       "      <td>178.658634</td>\n",
       "      <td>62.721379</td>\n",
       "      <td>55.951204</td>\n",
       "      <td>...</td>\n",
       "      <td>162.402114</td>\n",
       "      <td>1.037451</td>\n",
       "      <td>144.370090</td>\n",
       "      <td>208.113864</td>\n",
       "      <td>14.262005</td>\n",
       "      <td>43.673788</td>\n",
       "      <td>74.932959</td>\n",
       "      <td>38.915743</td>\n",
       "      <td>45.162795</td>\n",
       "      <td>79.388883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1002967_3</th>\n",
       "      <td>27.341552</td>\n",
       "      <td>120.232326</td>\n",
       "      <td>349.095667</td>\n",
       "      <td>193.395062</td>\n",
       "      <td>48.804329</td>\n",
       "      <td>53.359054</td>\n",
       "      <td>47.009773</td>\n",
       "      <td>152.000113</td>\n",
       "      <td>58.934078</td>\n",
       "      <td>43.200448</td>\n",
       "      <td>...</td>\n",
       "      <td>87.707173</td>\n",
       "      <td>9.256106</td>\n",
       "      <td>59.617181</td>\n",
       "      <td>247.086797</td>\n",
       "      <td>24.663200</td>\n",
       "      <td>51.850043</td>\n",
       "      <td>4.419552</td>\n",
       "      <td>27.218740</td>\n",
       "      <td>21.916938</td>\n",
       "      <td>68.751905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1003965_10</th>\n",
       "      <td>24.376980</td>\n",
       "      <td>172.249152</td>\n",
       "      <td>150.505049</td>\n",
       "      <td>18.110006</td>\n",
       "      <td>132.123840</td>\n",
       "      <td>28.703120</td>\n",
       "      <td>8.149837</td>\n",
       "      <td>115.727792</td>\n",
       "      <td>21.424122</td>\n",
       "      <td>16.858031</td>\n",
       "      <td>...</td>\n",
       "      <td>57.223006</td>\n",
       "      <td>41.482728</td>\n",
       "      <td>204.751568</td>\n",
       "      <td>35.337095</td>\n",
       "      <td>53.272303</td>\n",
       "      <td>127.598116</td>\n",
       "      <td>74.795004</td>\n",
       "      <td>93.190566</td>\n",
       "      <td>92.367805</td>\n",
       "      <td>104.356005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1004041_1</th>\n",
       "      <td>53.284447</td>\n",
       "      <td>0.306497</td>\n",
       "      <td>4.638289</td>\n",
       "      <td>61.191568</td>\n",
       "      <td>203.231975</td>\n",
       "      <td>88.284329</td>\n",
       "      <td>3.046813</td>\n",
       "      <td>5.299889</td>\n",
       "      <td>195.578089</td>\n",
       "      <td>6.252366</td>\n",
       "      <td>...</td>\n",
       "      <td>130.641019</td>\n",
       "      <td>9.193953</td>\n",
       "      <td>4.496304</td>\n",
       "      <td>142.587605</td>\n",
       "      <td>0.826234</td>\n",
       "      <td>266.520335</td>\n",
       "      <td>8.812933</td>\n",
       "      <td>52.492953</td>\n",
       "      <td>17.674539</td>\n",
       "      <td>40.778811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1004623_1</th>\n",
       "      <td>119.368003</td>\n",
       "      <td>33.717070</td>\n",
       "      <td>12.766574</td>\n",
       "      <td>38.006418</td>\n",
       "      <td>117.899451</td>\n",
       "      <td>185.485691</td>\n",
       "      <td>446.934636</td>\n",
       "      <td>11.686298</td>\n",
       "      <td>182.239036</td>\n",
       "      <td>24.190626</td>\n",
       "      <td>...</td>\n",
       "      <td>33.647304</td>\n",
       "      <td>118.834157</td>\n",
       "      <td>46.292226</td>\n",
       "      <td>35.476515</td>\n",
       "      <td>33.872342</td>\n",
       "      <td>52.061963</td>\n",
       "      <td>36.413021</td>\n",
       "      <td>111.688583</td>\n",
       "      <td>22.764500</td>\n",
       "      <td>39.151955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1005212_10</th>\n",
       "      <td>54.244065</td>\n",
       "      <td>267.630639</td>\n",
       "      <td>62.279420</td>\n",
       "      <td>124.831251</td>\n",
       "      <td>56.083609</td>\n",
       "      <td>54.110338</td>\n",
       "      <td>70.235402</td>\n",
       "      <td>175.656809</td>\n",
       "      <td>51.662677</td>\n",
       "      <td>47.703592</td>\n",
       "      <td>...</td>\n",
       "      <td>95.664787</td>\n",
       "      <td>7.083284</td>\n",
       "      <td>126.750742</td>\n",
       "      <td>115.386956</td>\n",
       "      <td>37.891127</td>\n",
       "      <td>82.778597</td>\n",
       "      <td>97.051830</td>\n",
       "      <td>151.377216</td>\n",
       "      <td>138.447806</td>\n",
       "      <td>85.003125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1005337_1</th>\n",
       "      <td>154.392649</td>\n",
       "      <td>83.607374</td>\n",
       "      <td>272.072922</td>\n",
       "      <td>63.092778</td>\n",
       "      <td>32.272982</td>\n",
       "      <td>24.834490</td>\n",
       "      <td>86.632449</td>\n",
       "      <td>394.377900</td>\n",
       "      <td>16.996699</td>\n",
       "      <td>14.575021</td>\n",
       "      <td>...</td>\n",
       "      <td>143.982645</td>\n",
       "      <td>1.291334</td>\n",
       "      <td>13.995447</td>\n",
       "      <td>84.466980</td>\n",
       "      <td>17.595855</td>\n",
       "      <td>26.110809</td>\n",
       "      <td>72.402559</td>\n",
       "      <td>19.235825</td>\n",
       "      <td>45.623446</td>\n",
       "      <td>87.443748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1005475_1</th>\n",
       "      <td>41.334795</td>\n",
       "      <td>52.794187</td>\n",
       "      <td>132.989227</td>\n",
       "      <td>34.285508</td>\n",
       "      <td>54.319255</td>\n",
       "      <td>87.355216</td>\n",
       "      <td>26.796499</td>\n",
       "      <td>131.341136</td>\n",
       "      <td>50.314613</td>\n",
       "      <td>108.054515</td>\n",
       "      <td>...</td>\n",
       "      <td>83.264222</td>\n",
       "      <td>81.462515</td>\n",
       "      <td>42.030277</td>\n",
       "      <td>165.868790</td>\n",
       "      <td>53.777821</td>\n",
       "      <td>67.575000</td>\n",
       "      <td>134.316490</td>\n",
       "      <td>44.658326</td>\n",
       "      <td>73.540616</td>\n",
       "      <td>53.925726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1005733_1</th>\n",
       "      <td>131.967098</td>\n",
       "      <td>23.004395</td>\n",
       "      <td>116.745592</td>\n",
       "      <td>46.300057</td>\n",
       "      <td>69.666192</td>\n",
       "      <td>45.045068</td>\n",
       "      <td>55.899728</td>\n",
       "      <td>2.445917</td>\n",
       "      <td>17.834656</td>\n",
       "      <td>61.272581</td>\n",
       "      <td>...</td>\n",
       "      <td>97.824029</td>\n",
       "      <td>10.591840</td>\n",
       "      <td>104.433287</td>\n",
       "      <td>103.352557</td>\n",
       "      <td>69.095019</td>\n",
       "      <td>48.348803</td>\n",
       "      <td>475.281984</td>\n",
       "      <td>13.203118</td>\n",
       "      <td>111.412930</td>\n",
       "      <td>141.809785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1006916_3</th>\n",
       "      <td>175.810806</td>\n",
       "      <td>0.139214</td>\n",
       "      <td>5.430354</td>\n",
       "      <td>25.455806</td>\n",
       "      <td>46.773192</td>\n",
       "      <td>102.757996</td>\n",
       "      <td>19.256817</td>\n",
       "      <td>3.847794</td>\n",
       "      <td>144.468301</td>\n",
       "      <td>113.894306</td>\n",
       "      <td>...</td>\n",
       "      <td>177.030833</td>\n",
       "      <td>10.818418</td>\n",
       "      <td>53.110737</td>\n",
       "      <td>106.636277</td>\n",
       "      <td>106.527928</td>\n",
       "      <td>50.580339</td>\n",
       "      <td>269.741368</td>\n",
       "      <td>20.338973</td>\n",
       "      <td>18.281009</td>\n",
       "      <td>82.846367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1006998_1</th>\n",
       "      <td>217.632768</td>\n",
       "      <td>32.915850</td>\n",
       "      <td>277.607949</td>\n",
       "      <td>32.080584</td>\n",
       "      <td>108.743595</td>\n",
       "      <td>71.300360</td>\n",
       "      <td>38.116813</td>\n",
       "      <td>100.084247</td>\n",
       "      <td>75.199448</td>\n",
       "      <td>98.662185</td>\n",
       "      <td>...</td>\n",
       "      <td>112.682155</td>\n",
       "      <td>341.052887</td>\n",
       "      <td>62.696106</td>\n",
       "      <td>14.234573</td>\n",
       "      <td>65.602727</td>\n",
       "      <td>137.541238</td>\n",
       "      <td>16.956252</td>\n",
       "      <td>105.403932</td>\n",
       "      <td>17.336581</td>\n",
       "      <td>22.917993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1007311_1</th>\n",
       "      <td>172.994521</td>\n",
       "      <td>119.471056</td>\n",
       "      <td>5.150463</td>\n",
       "      <td>20.349668</td>\n",
       "      <td>69.172643</td>\n",
       "      <td>3.112321</td>\n",
       "      <td>29.750006</td>\n",
       "      <td>7.817663</td>\n",
       "      <td>87.060368</td>\n",
       "      <td>219.169475</td>\n",
       "      <td>...</td>\n",
       "      <td>83.179488</td>\n",
       "      <td>287.281548</td>\n",
       "      <td>66.359367</td>\n",
       "      <td>156.395535</td>\n",
       "      <td>43.959460</td>\n",
       "      <td>52.131438</td>\n",
       "      <td>228.362410</td>\n",
       "      <td>26.485740</td>\n",
       "      <td>196.476485</td>\n",
       "      <td>78.682065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1007631_1</th>\n",
       "      <td>36.864080</td>\n",
       "      <td>117.667665</td>\n",
       "      <td>216.172100</td>\n",
       "      <td>126.561557</td>\n",
       "      <td>29.536866</td>\n",
       "      <td>148.119197</td>\n",
       "      <td>106.115331</td>\n",
       "      <td>173.137766</td>\n",
       "      <td>10.344472</td>\n",
       "      <td>68.278372</td>\n",
       "      <td>...</td>\n",
       "      <td>26.826830</td>\n",
       "      <td>61.329672</td>\n",
       "      <td>65.778732</td>\n",
       "      <td>24.884930</td>\n",
       "      <td>7.335209</td>\n",
       "      <td>9.038559</td>\n",
       "      <td>46.606115</td>\n",
       "      <td>1128.870371</td>\n",
       "      <td>15.314261</td>\n",
       "      <td>9.883403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1007659_2</th>\n",
       "      <td>323.951407</td>\n",
       "      <td>2.216117</td>\n",
       "      <td>11.720689</td>\n",
       "      <td>30.269739</td>\n",
       "      <td>3.051028</td>\n",
       "      <td>530.322777</td>\n",
       "      <td>41.138164</td>\n",
       "      <td>10.867847</td>\n",
       "      <td>38.951188</td>\n",
       "      <td>18.246542</td>\n",
       "      <td>...</td>\n",
       "      <td>187.940755</td>\n",
       "      <td>22.869324</td>\n",
       "      <td>14.214832</td>\n",
       "      <td>134.948844</td>\n",
       "      <td>327.872525</td>\n",
       "      <td>62.831065</td>\n",
       "      <td>302.385864</td>\n",
       "      <td>24.794732</td>\n",
       "      <td>7.176967</td>\n",
       "      <td>193.394550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_100772_1</th>\n",
       "      <td>221.756668</td>\n",
       "      <td>507.788938</td>\n",
       "      <td>5.102968</td>\n",
       "      <td>103.101741</td>\n",
       "      <td>195.245184</td>\n",
       "      <td>24.812102</td>\n",
       "      <td>62.937204</td>\n",
       "      <td>5.882766</td>\n",
       "      <td>161.211253</td>\n",
       "      <td>73.688159</td>\n",
       "      <td>...</td>\n",
       "      <td>95.280290</td>\n",
       "      <td>34.031023</td>\n",
       "      <td>139.726011</td>\n",
       "      <td>58.911694</td>\n",
       "      <td>72.156834</td>\n",
       "      <td>48.211895</td>\n",
       "      <td>373.702282</td>\n",
       "      <td>23.264329</td>\n",
       "      <td>121.162355</td>\n",
       "      <td>335.377411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1007949_1</th>\n",
       "      <td>62.968256</td>\n",
       "      <td>58.710044</td>\n",
       "      <td>281.423659</td>\n",
       "      <td>22.513453</td>\n",
       "      <td>51.175672</td>\n",
       "      <td>125.605334</td>\n",
       "      <td>37.996999</td>\n",
       "      <td>264.320399</td>\n",
       "      <td>33.884970</td>\n",
       "      <td>116.922834</td>\n",
       "      <td>...</td>\n",
       "      <td>79.150747</td>\n",
       "      <td>2.793514</td>\n",
       "      <td>50.310957</td>\n",
       "      <td>222.896789</td>\n",
       "      <td>85.199731</td>\n",
       "      <td>49.285397</td>\n",
       "      <td>178.029218</td>\n",
       "      <td>31.586833</td>\n",
       "      <td>68.288730</td>\n",
       "      <td>83.179259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008105_1</th>\n",
       "      <td>97.160518</td>\n",
       "      <td>53.102530</td>\n",
       "      <td>46.897418</td>\n",
       "      <td>140.513848</td>\n",
       "      <td>35.394027</td>\n",
       "      <td>120.324332</td>\n",
       "      <td>244.362456</td>\n",
       "      <td>602.312953</td>\n",
       "      <td>68.507910</td>\n",
       "      <td>10.580655</td>\n",
       "      <td>...</td>\n",
       "      <td>48.559364</td>\n",
       "      <td>58.858740</td>\n",
       "      <td>57.157386</td>\n",
       "      <td>11.490327</td>\n",
       "      <td>42.118520</td>\n",
       "      <td>23.934693</td>\n",
       "      <td>35.413779</td>\n",
       "      <td>30.884223</td>\n",
       "      <td>153.488932</td>\n",
       "      <td>30.721960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008105_2</th>\n",
       "      <td>72.357921</td>\n",
       "      <td>20.590128</td>\n",
       "      <td>19.745163</td>\n",
       "      <td>43.018586</td>\n",
       "      <td>39.804201</td>\n",
       "      <td>79.228430</td>\n",
       "      <td>94.875345</td>\n",
       "      <td>747.386813</td>\n",
       "      <td>60.187154</td>\n",
       "      <td>7.833829</td>\n",
       "      <td>...</td>\n",
       "      <td>43.691513</td>\n",
       "      <td>51.254832</td>\n",
       "      <td>64.158146</td>\n",
       "      <td>20.654935</td>\n",
       "      <td>42.772805</td>\n",
       "      <td>19.853697</td>\n",
       "      <td>49.171634</td>\n",
       "      <td>345.129520</td>\n",
       "      <td>155.586749</td>\n",
       "      <td>30.769563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008105_3</th>\n",
       "      <td>233.355649</td>\n",
       "      <td>74.324369</td>\n",
       "      <td>15.008242</td>\n",
       "      <td>27.759816</td>\n",
       "      <td>68.564973</td>\n",
       "      <td>74.368965</td>\n",
       "      <td>43.459606</td>\n",
       "      <td>353.385122</td>\n",
       "      <td>33.754965</td>\n",
       "      <td>7.039364</td>\n",
       "      <td>...</td>\n",
       "      <td>31.286745</td>\n",
       "      <td>33.925599</td>\n",
       "      <td>39.628948</td>\n",
       "      <td>22.844824</td>\n",
       "      <td>41.541536</td>\n",
       "      <td>22.624244</td>\n",
       "      <td>51.710949</td>\n",
       "      <td>28.625361</td>\n",
       "      <td>164.788802</td>\n",
       "      <td>13.913015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008180_1</th>\n",
       "      <td>15.187049</td>\n",
       "      <td>113.536625</td>\n",
       "      <td>13.021212</td>\n",
       "      <td>61.337747</td>\n",
       "      <td>102.489074</td>\n",
       "      <td>26.471033</td>\n",
       "      <td>223.678836</td>\n",
       "      <td>39.319820</td>\n",
       "      <td>47.393270</td>\n",
       "      <td>94.291313</td>\n",
       "      <td>...</td>\n",
       "      <td>73.628504</td>\n",
       "      <td>291.178040</td>\n",
       "      <td>50.062576</td>\n",
       "      <td>27.931189</td>\n",
       "      <td>56.354810</td>\n",
       "      <td>26.735054</td>\n",
       "      <td>235.855610</td>\n",
       "      <td>111.598174</td>\n",
       "      <td>28.519930</td>\n",
       "      <td>45.642919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008218_1</th>\n",
       "      <td>56.584412</td>\n",
       "      <td>138.779130</td>\n",
       "      <td>242.561476</td>\n",
       "      <td>15.886922</td>\n",
       "      <td>69.635360</td>\n",
       "      <td>9.417897</td>\n",
       "      <td>55.037163</td>\n",
       "      <td>54.076108</td>\n",
       "      <td>31.608293</td>\n",
       "      <td>22.455110</td>\n",
       "      <td>...</td>\n",
       "      <td>121.542235</td>\n",
       "      <td>16.536745</td>\n",
       "      <td>22.638705</td>\n",
       "      <td>138.888164</td>\n",
       "      <td>64.707997</td>\n",
       "      <td>57.914429</td>\n",
       "      <td>171.654608</td>\n",
       "      <td>36.243821</td>\n",
       "      <td>40.396772</td>\n",
       "      <td>119.941563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008435_3</th>\n",
       "      <td>36.968736</td>\n",
       "      <td>173.767161</td>\n",
       "      <td>95.189930</td>\n",
       "      <td>127.752392</td>\n",
       "      <td>24.688385</td>\n",
       "      <td>187.116384</td>\n",
       "      <td>40.011542</td>\n",
       "      <td>101.439969</td>\n",
       "      <td>23.529278</td>\n",
       "      <td>19.802737</td>\n",
       "      <td>...</td>\n",
       "      <td>37.230945</td>\n",
       "      <td>264.905214</td>\n",
       "      <td>40.525079</td>\n",
       "      <td>116.844388</td>\n",
       "      <td>37.470565</td>\n",
       "      <td>51.387261</td>\n",
       "      <td>140.366705</td>\n",
       "      <td>61.901285</td>\n",
       "      <td>91.628428</td>\n",
       "      <td>44.829909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008548_2</th>\n",
       "      <td>114.743947</td>\n",
       "      <td>44.819853</td>\n",
       "      <td>65.549676</td>\n",
       "      <td>24.236662</td>\n",
       "      <td>103.924199</td>\n",
       "      <td>105.443975</td>\n",
       "      <td>25.129432</td>\n",
       "      <td>34.657164</td>\n",
       "      <td>96.906225</td>\n",
       "      <td>24.770467</td>\n",
       "      <td>...</td>\n",
       "      <td>42.855305</td>\n",
       "      <td>150.394764</td>\n",
       "      <td>63.493967</td>\n",
       "      <td>184.482410</td>\n",
       "      <td>42.900626</td>\n",
       "      <td>122.452632</td>\n",
       "      <td>195.462071</td>\n",
       "      <td>47.728523</td>\n",
       "      <td>131.932277</td>\n",
       "      <td>58.422157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008597_2</th>\n",
       "      <td>336.860031</td>\n",
       "      <td>10.450378</td>\n",
       "      <td>12.066680</td>\n",
       "      <td>51.704662</td>\n",
       "      <td>66.516863</td>\n",
       "      <td>121.469664</td>\n",
       "      <td>109.961704</td>\n",
       "      <td>83.462251</td>\n",
       "      <td>3.592345</td>\n",
       "      <td>89.744451</td>\n",
       "      <td>...</td>\n",
       "      <td>262.603783</td>\n",
       "      <td>0.864858</td>\n",
       "      <td>175.382372</td>\n",
       "      <td>25.034425</td>\n",
       "      <td>71.264266</td>\n",
       "      <td>55.105001</td>\n",
       "      <td>606.904975</td>\n",
       "      <td>7.139729</td>\n",
       "      <td>193.304802</td>\n",
       "      <td>42.680944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008705_2</th>\n",
       "      <td>83.383882</td>\n",
       "      <td>53.365558</td>\n",
       "      <td>47.799088</td>\n",
       "      <td>58.974744</td>\n",
       "      <td>92.913103</td>\n",
       "      <td>61.585567</td>\n",
       "      <td>161.752034</td>\n",
       "      <td>24.575543</td>\n",
       "      <td>97.624925</td>\n",
       "      <td>114.127828</td>\n",
       "      <td>...</td>\n",
       "      <td>85.751608</td>\n",
       "      <td>132.415589</td>\n",
       "      <td>66.312452</td>\n",
       "      <td>120.764616</td>\n",
       "      <td>95.294835</td>\n",
       "      <td>185.325284</td>\n",
       "      <td>127.312305</td>\n",
       "      <td>71.008522</td>\n",
       "      <td>72.224352</td>\n",
       "      <td>65.413707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_1008991_1</th>\n",
       "      <td>14.183006</td>\n",
       "      <td>21.144467</td>\n",
       "      <td>16.343396</td>\n",
       "      <td>54.560932</td>\n",
       "      <td>27.145712</td>\n",
       "      <td>248.087044</td>\n",
       "      <td>20.066131</td>\n",
       "      <td>10.252830</td>\n",
       "      <td>239.079743</td>\n",
       "      <td>482.507133</td>\n",
       "      <td>...</td>\n",
       "      <td>4.575419</td>\n",
       "      <td>2.029517</td>\n",
       "      <td>24.949074</td>\n",
       "      <td>22.483890</td>\n",
       "      <td>275.678256</td>\n",
       "      <td>45.283593</td>\n",
       "      <td>43.438743</td>\n",
       "      <td>24.628750</td>\n",
       "      <td>7.154000</td>\n",
       "      <td>285.272588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_986881_2</th>\n",
       "      <td>71.725067</td>\n",
       "      <td>46.142810</td>\n",
       "      <td>249.027866</td>\n",
       "      <td>20.509308</td>\n",
       "      <td>63.446196</td>\n",
       "      <td>63.527754</td>\n",
       "      <td>24.583542</td>\n",
       "      <td>96.667323</td>\n",
       "      <td>52.329958</td>\n",
       "      <td>64.337782</td>\n",
       "      <td>...</td>\n",
       "      <td>64.452276</td>\n",
       "      <td>158.935581</td>\n",
       "      <td>65.364268</td>\n",
       "      <td>152.094888</td>\n",
       "      <td>47.966426</td>\n",
       "      <td>60.565922</td>\n",
       "      <td>94.545120</td>\n",
       "      <td>28.180323</td>\n",
       "      <td>121.394564</td>\n",
       "      <td>60.364492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_988185_3</th>\n",
       "      <td>284.722747</td>\n",
       "      <td>109.255008</td>\n",
       "      <td>21.199096</td>\n",
       "      <td>113.321378</td>\n",
       "      <td>118.419744</td>\n",
       "      <td>80.363294</td>\n",
       "      <td>42.946036</td>\n",
       "      <td>464.331237</td>\n",
       "      <td>49.561183</td>\n",
       "      <td>6.171331</td>\n",
       "      <td>...</td>\n",
       "      <td>23.416377</td>\n",
       "      <td>54.173183</td>\n",
       "      <td>37.626349</td>\n",
       "      <td>106.743917</td>\n",
       "      <td>38.127006</td>\n",
       "      <td>65.221674</td>\n",
       "      <td>50.037052</td>\n",
       "      <td>25.940555</td>\n",
       "      <td>139.556174</td>\n",
       "      <td>16.241541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_989559_1</th>\n",
       "      <td>53.745577</td>\n",
       "      <td>285.744149</td>\n",
       "      <td>109.222848</td>\n",
       "      <td>30.145157</td>\n",
       "      <td>59.419067</td>\n",
       "      <td>101.868692</td>\n",
       "      <td>91.933955</td>\n",
       "      <td>199.032317</td>\n",
       "      <td>124.424764</td>\n",
       "      <td>75.420037</td>\n",
       "      <td>...</td>\n",
       "      <td>191.374727</td>\n",
       "      <td>27.189949</td>\n",
       "      <td>52.427279</td>\n",
       "      <td>87.246854</td>\n",
       "      <td>48.851018</td>\n",
       "      <td>64.941353</td>\n",
       "      <td>33.978277</td>\n",
       "      <td>12.991661</td>\n",
       "      <td>62.919910</td>\n",
       "      <td>11.676106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_989607_1</th>\n",
       "      <td>21.372486</td>\n",
       "      <td>5.238690</td>\n",
       "      <td>4.239787</td>\n",
       "      <td>7.797743</td>\n",
       "      <td>17.208427</td>\n",
       "      <td>84.646844</td>\n",
       "      <td>10.535081</td>\n",
       "      <td>57.181267</td>\n",
       "      <td>4.448012</td>\n",
       "      <td>40.092313</td>\n",
       "      <td>...</td>\n",
       "      <td>6.967357</td>\n",
       "      <td>1243.113262</td>\n",
       "      <td>22.678447</td>\n",
       "      <td>541.383898</td>\n",
       "      <td>38.994140</td>\n",
       "      <td>39.977930</td>\n",
       "      <td>163.316391</td>\n",
       "      <td>53.464826</td>\n",
       "      <td>84.602661</td>\n",
       "      <td>16.568652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_989714_5</th>\n",
       "      <td>18.102763</td>\n",
       "      <td>33.725137</td>\n",
       "      <td>78.541373</td>\n",
       "      <td>64.782197</td>\n",
       "      <td>25.561647</td>\n",
       "      <td>143.368810</td>\n",
       "      <td>28.542558</td>\n",
       "      <td>24.711359</td>\n",
       "      <td>171.413633</td>\n",
       "      <td>280.820581</td>\n",
       "      <td>...</td>\n",
       "      <td>41.318265</td>\n",
       "      <td>173.799895</td>\n",
       "      <td>46.810370</td>\n",
       "      <td>229.354717</td>\n",
       "      <td>138.120506</td>\n",
       "      <td>181.006556</td>\n",
       "      <td>99.057888</td>\n",
       "      <td>58.995529</td>\n",
       "      <td>82.048609</td>\n",
       "      <td>144.450007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_989874_5</th>\n",
       "      <td>436.717957</td>\n",
       "      <td>7.261263</td>\n",
       "      <td>35.282143</td>\n",
       "      <td>97.395999</td>\n",
       "      <td>39.009174</td>\n",
       "      <td>248.418095</td>\n",
       "      <td>40.658572</td>\n",
       "      <td>23.025467</td>\n",
       "      <td>191.905049</td>\n",
       "      <td>6.042956</td>\n",
       "      <td>...</td>\n",
       "      <td>79.741161</td>\n",
       "      <td>19.548117</td>\n",
       "      <td>89.645424</td>\n",
       "      <td>18.464332</td>\n",
       "      <td>431.256857</td>\n",
       "      <td>54.006182</td>\n",
       "      <td>28.161478</td>\n",
       "      <td>48.214084</td>\n",
       "      <td>59.162636</td>\n",
       "      <td>15.641727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_990427_1</th>\n",
       "      <td>88.757610</td>\n",
       "      <td>70.961000</td>\n",
       "      <td>151.333799</td>\n",
       "      <td>36.360445</td>\n",
       "      <td>140.097476</td>\n",
       "      <td>6.625953</td>\n",
       "      <td>45.700091</td>\n",
       "      <td>58.809509</td>\n",
       "      <td>40.696426</td>\n",
       "      <td>7.899579</td>\n",
       "      <td>...</td>\n",
       "      <td>110.116921</td>\n",
       "      <td>10.819953</td>\n",
       "      <td>34.073402</td>\n",
       "      <td>13.432432</td>\n",
       "      <td>47.333795</td>\n",
       "      <td>149.829822</td>\n",
       "      <td>327.653439</td>\n",
       "      <td>62.410996</td>\n",
       "      <td>14.494765</td>\n",
       "      <td>90.445981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_990682_2</th>\n",
       "      <td>3.124727</td>\n",
       "      <td>76.672831</td>\n",
       "      <td>102.157108</td>\n",
       "      <td>59.000290</td>\n",
       "      <td>30.144200</td>\n",
       "      <td>82.896231</td>\n",
       "      <td>10.458825</td>\n",
       "      <td>13.273222</td>\n",
       "      <td>125.100109</td>\n",
       "      <td>0.376024</td>\n",
       "      <td>...</td>\n",
       "      <td>83.164092</td>\n",
       "      <td>111.125592</td>\n",
       "      <td>30.983864</td>\n",
       "      <td>24.564292</td>\n",
       "      <td>159.006282</td>\n",
       "      <td>30.446200</td>\n",
       "      <td>121.973523</td>\n",
       "      <td>411.428478</td>\n",
       "      <td>163.663505</td>\n",
       "      <td>56.555794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_991061_1</th>\n",
       "      <td>36.536070</td>\n",
       "      <td>25.212403</td>\n",
       "      <td>200.306933</td>\n",
       "      <td>25.807963</td>\n",
       "      <td>47.756136</td>\n",
       "      <td>57.971996</td>\n",
       "      <td>269.991030</td>\n",
       "      <td>129.861009</td>\n",
       "      <td>41.787830</td>\n",
       "      <td>72.995633</td>\n",
       "      <td>...</td>\n",
       "      <td>63.342419</td>\n",
       "      <td>94.161809</td>\n",
       "      <td>51.982227</td>\n",
       "      <td>326.181419</td>\n",
       "      <td>37.302418</td>\n",
       "      <td>38.456098</td>\n",
       "      <td>146.162365</td>\n",
       "      <td>15.851840</td>\n",
       "      <td>80.470856</td>\n",
       "      <td>57.616524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_991419_1</th>\n",
       "      <td>3.012734</td>\n",
       "      <td>0.592461</td>\n",
       "      <td>2.226710</td>\n",
       "      <td>23.017513</td>\n",
       "      <td>3.387003</td>\n",
       "      <td>1.712442</td>\n",
       "      <td>22.779307</td>\n",
       "      <td>1.775004</td>\n",
       "      <td>333.171941</td>\n",
       "      <td>182.074637</td>\n",
       "      <td>...</td>\n",
       "      <td>13.904292</td>\n",
       "      <td>6.770838</td>\n",
       "      <td>19.667648</td>\n",
       "      <td>113.029613</td>\n",
       "      <td>102.713870</td>\n",
       "      <td>187.718389</td>\n",
       "      <td>354.255053</td>\n",
       "      <td>84.073836</td>\n",
       "      <td>11.267959</td>\n",
       "      <td>11.200960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_991741_1</th>\n",
       "      <td>36.078577</td>\n",
       "      <td>6.200558</td>\n",
       "      <td>11.098007</td>\n",
       "      <td>16.391302</td>\n",
       "      <td>6.033534</td>\n",
       "      <td>6.282880</td>\n",
       "      <td>29.432023</td>\n",
       "      <td>1.379454</td>\n",
       "      <td>12.175121</td>\n",
       "      <td>756.155606</td>\n",
       "      <td>...</td>\n",
       "      <td>24.618277</td>\n",
       "      <td>24.585423</td>\n",
       "      <td>78.446327</td>\n",
       "      <td>161.702029</td>\n",
       "      <td>28.955127</td>\n",
       "      <td>96.863843</td>\n",
       "      <td>6.997331</td>\n",
       "      <td>12.687838</td>\n",
       "      <td>28.004871</td>\n",
       "      <td>72.402798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_991826_1</th>\n",
       "      <td>8.093804</td>\n",
       "      <td>52.305888</td>\n",
       "      <td>117.718034</td>\n",
       "      <td>58.529716</td>\n",
       "      <td>47.984053</td>\n",
       "      <td>85.682910</td>\n",
       "      <td>12.752715</td>\n",
       "      <td>13.151539</td>\n",
       "      <td>75.009943</td>\n",
       "      <td>27.793513</td>\n",
       "      <td>...</td>\n",
       "      <td>48.497533</td>\n",
       "      <td>178.090202</td>\n",
       "      <td>58.703498</td>\n",
       "      <td>17.100617</td>\n",
       "      <td>44.996199</td>\n",
       "      <td>28.637388</td>\n",
       "      <td>105.409735</td>\n",
       "      <td>400.144006</td>\n",
       "      <td>258.115744</td>\n",
       "      <td>49.954926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_991826_3</th>\n",
       "      <td>19.348388</td>\n",
       "      <td>15.955913</td>\n",
       "      <td>54.466123</td>\n",
       "      <td>38.363492</td>\n",
       "      <td>23.794290</td>\n",
       "      <td>51.280761</td>\n",
       "      <td>54.685985</td>\n",
       "      <td>13.570566</td>\n",
       "      <td>169.254131</td>\n",
       "      <td>187.965419</td>\n",
       "      <td>...</td>\n",
       "      <td>45.318710</td>\n",
       "      <td>190.781121</td>\n",
       "      <td>140.921684</td>\n",
       "      <td>580.472026</td>\n",
       "      <td>15.913658</td>\n",
       "      <td>23.240362</td>\n",
       "      <td>32.949575</td>\n",
       "      <td>130.358814</td>\n",
       "      <td>59.896302</td>\n",
       "      <td>21.846505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_99188_2</th>\n",
       "      <td>40.975287</td>\n",
       "      <td>35.414517</td>\n",
       "      <td>17.301511</td>\n",
       "      <td>92.112643</td>\n",
       "      <td>411.046092</td>\n",
       "      <td>59.960873</td>\n",
       "      <td>84.538943</td>\n",
       "      <td>114.003832</td>\n",
       "      <td>25.825147</td>\n",
       "      <td>69.377144</td>\n",
       "      <td>...</td>\n",
       "      <td>60.121992</td>\n",
       "      <td>241.589395</td>\n",
       "      <td>43.355378</td>\n",
       "      <td>112.401977</td>\n",
       "      <td>43.606543</td>\n",
       "      <td>42.533944</td>\n",
       "      <td>101.070831</td>\n",
       "      <td>38.945893</td>\n",
       "      <td>258.366007</td>\n",
       "      <td>24.085724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_992336_2</th>\n",
       "      <td>69.489179</td>\n",
       "      <td>110.334996</td>\n",
       "      <td>47.178478</td>\n",
       "      <td>0.706841</td>\n",
       "      <td>53.415820</td>\n",
       "      <td>110.156962</td>\n",
       "      <td>181.152375</td>\n",
       "      <td>82.686340</td>\n",
       "      <td>52.833980</td>\n",
       "      <td>116.850245</td>\n",
       "      <td>...</td>\n",
       "      <td>81.585943</td>\n",
       "      <td>0.860196</td>\n",
       "      <td>68.124300</td>\n",
       "      <td>15.599012</td>\n",
       "      <td>49.605158</td>\n",
       "      <td>164.542825</td>\n",
       "      <td>30.371297</td>\n",
       "      <td>525.937837</td>\n",
       "      <td>35.865205</td>\n",
       "      <td>66.438763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_992839_2</th>\n",
       "      <td>41.191678</td>\n",
       "      <td>236.993691</td>\n",
       "      <td>582.003353</td>\n",
       "      <td>26.288025</td>\n",
       "      <td>83.354495</td>\n",
       "      <td>582.233477</td>\n",
       "      <td>87.958774</td>\n",
       "      <td>83.684066</td>\n",
       "      <td>71.018943</td>\n",
       "      <td>18.976264</td>\n",
       "      <td>...</td>\n",
       "      <td>146.727643</td>\n",
       "      <td>66.717530</td>\n",
       "      <td>56.427479</td>\n",
       "      <td>28.374838</td>\n",
       "      <td>344.149848</td>\n",
       "      <td>52.515811</td>\n",
       "      <td>236.883788</td>\n",
       "      <td>130.909592</td>\n",
       "      <td>123.933795</td>\n",
       "      <td>102.075627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_99300_10</th>\n",
       "      <td>57.533476</td>\n",
       "      <td>51.688237</td>\n",
       "      <td>70.426056</td>\n",
       "      <td>218.373047</td>\n",
       "      <td>1001.555339</td>\n",
       "      <td>63.822298</td>\n",
       "      <td>52.255680</td>\n",
       "      <td>60.389625</td>\n",
       "      <td>109.716554</td>\n",
       "      <td>145.773820</td>\n",
       "      <td>...</td>\n",
       "      <td>102.388925</td>\n",
       "      <td>284.636193</td>\n",
       "      <td>31.342840</td>\n",
       "      <td>146.327508</td>\n",
       "      <td>34.910865</td>\n",
       "      <td>62.307465</td>\n",
       "      <td>128.704373</td>\n",
       "      <td>53.509529</td>\n",
       "      <td>110.656657</td>\n",
       "      <td>47.255105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_993197_4</th>\n",
       "      <td>7.388702</td>\n",
       "      <td>18.812668</td>\n",
       "      <td>3.064288</td>\n",
       "      <td>11.812190</td>\n",
       "      <td>16.704778</td>\n",
       "      <td>44.678333</td>\n",
       "      <td>30.332263</td>\n",
       "      <td>1.236302</td>\n",
       "      <td>30.071025</td>\n",
       "      <td>91.500718</td>\n",
       "      <td>...</td>\n",
       "      <td>53.954294</td>\n",
       "      <td>13.450632</td>\n",
       "      <td>14.947777</td>\n",
       "      <td>8.502076</td>\n",
       "      <td>32.254392</td>\n",
       "      <td>18.588861</td>\n",
       "      <td>13.478309</td>\n",
       "      <td>615.029349</td>\n",
       "      <td>337.343961</td>\n",
       "      <td>16.132682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_993759_8</th>\n",
       "      <td>72.136854</td>\n",
       "      <td>45.623568</td>\n",
       "      <td>39.880100</td>\n",
       "      <td>42.682101</td>\n",
       "      <td>40.751973</td>\n",
       "      <td>19.616386</td>\n",
       "      <td>98.518508</td>\n",
       "      <td>344.930759</td>\n",
       "      <td>85.579857</td>\n",
       "      <td>39.813719</td>\n",
       "      <td>...</td>\n",
       "      <td>4.379129</td>\n",
       "      <td>4.373121</td>\n",
       "      <td>149.948053</td>\n",
       "      <td>53.937557</td>\n",
       "      <td>208.164519</td>\n",
       "      <td>97.324389</td>\n",
       "      <td>7.107926</td>\n",
       "      <td>53.071577</td>\n",
       "      <td>2.960770</td>\n",
       "      <td>6.403968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_9941_1</th>\n",
       "      <td>421.955852</td>\n",
       "      <td>70.549217</td>\n",
       "      <td>30.678601</td>\n",
       "      <td>118.981610</td>\n",
       "      <td>70.286791</td>\n",
       "      <td>69.750482</td>\n",
       "      <td>145.949478</td>\n",
       "      <td>244.268993</td>\n",
       "      <td>21.943717</td>\n",
       "      <td>14.361836</td>\n",
       "      <td>...</td>\n",
       "      <td>20.626701</td>\n",
       "      <td>62.686112</td>\n",
       "      <td>34.038831</td>\n",
       "      <td>64.669907</td>\n",
       "      <td>73.436938</td>\n",
       "      <td>77.250724</td>\n",
       "      <td>155.156430</td>\n",
       "      <td>27.444070</td>\n",
       "      <td>118.640369</td>\n",
       "      <td>34.738022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_99441_1</th>\n",
       "      <td>41.289352</td>\n",
       "      <td>93.724856</td>\n",
       "      <td>64.446688</td>\n",
       "      <td>300.902263</td>\n",
       "      <td>1564.878900</td>\n",
       "      <td>11.398853</td>\n",
       "      <td>119.783493</td>\n",
       "      <td>34.885255</td>\n",
       "      <td>3.385661</td>\n",
       "      <td>3.022282</td>\n",
       "      <td>...</td>\n",
       "      <td>1.648258</td>\n",
       "      <td>1.586085</td>\n",
       "      <td>1338.174649</td>\n",
       "      <td>6.760904</td>\n",
       "      <td>28.920598</td>\n",
       "      <td>6.678438</td>\n",
       "      <td>14.575007</td>\n",
       "      <td>50.024170</td>\n",
       "      <td>7.771759</td>\n",
       "      <td>8.543831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_99524_2</th>\n",
       "      <td>778.730579</td>\n",
       "      <td>11.520877</td>\n",
       "      <td>19.098173</td>\n",
       "      <td>40.797132</td>\n",
       "      <td>407.020702</td>\n",
       "      <td>2.038803</td>\n",
       "      <td>72.388909</td>\n",
       "      <td>11.173429</td>\n",
       "      <td>15.710797</td>\n",
       "      <td>6.437326</td>\n",
       "      <td>...</td>\n",
       "      <td>3.160433</td>\n",
       "      <td>5.150197</td>\n",
       "      <td>24.043758</td>\n",
       "      <td>120.281227</td>\n",
       "      <td>34.992531</td>\n",
       "      <td>96.598168</td>\n",
       "      <td>66.783737</td>\n",
       "      <td>91.903437</td>\n",
       "      <td>114.752923</td>\n",
       "      <td>16.189056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_995394_1</th>\n",
       "      <td>88.744787</td>\n",
       "      <td>6.591762</td>\n",
       "      <td>184.875953</td>\n",
       "      <td>81.745859</td>\n",
       "      <td>46.235319</td>\n",
       "      <td>103.827076</td>\n",
       "      <td>67.934138</td>\n",
       "      <td>29.360562</td>\n",
       "      <td>82.055675</td>\n",
       "      <td>132.537694</td>\n",
       "      <td>...</td>\n",
       "      <td>141.450007</td>\n",
       "      <td>199.246303</td>\n",
       "      <td>74.208047</td>\n",
       "      <td>102.196755</td>\n",
       "      <td>104.131953</td>\n",
       "      <td>116.885412</td>\n",
       "      <td>39.239192</td>\n",
       "      <td>66.832456</td>\n",
       "      <td>99.410804</td>\n",
       "      <td>122.105325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_995896_4</th>\n",
       "      <td>141.243431</td>\n",
       "      <td>107.025153</td>\n",
       "      <td>141.962239</td>\n",
       "      <td>20.507947</td>\n",
       "      <td>112.438742</td>\n",
       "      <td>13.333790</td>\n",
       "      <td>141.704716</td>\n",
       "      <td>42.099626</td>\n",
       "      <td>71.641031</td>\n",
       "      <td>76.769254</td>\n",
       "      <td>...</td>\n",
       "      <td>62.294905</td>\n",
       "      <td>174.828352</td>\n",
       "      <td>145.722862</td>\n",
       "      <td>35.921944</td>\n",
       "      <td>44.932540</td>\n",
       "      <td>193.934256</td>\n",
       "      <td>26.622678</td>\n",
       "      <td>38.868802</td>\n",
       "      <td>65.478220</td>\n",
       "      <td>138.665934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_996093_4</th>\n",
       "      <td>41.399251</td>\n",
       "      <td>64.807983</td>\n",
       "      <td>61.345281</td>\n",
       "      <td>80.127394</td>\n",
       "      <td>117.044491</td>\n",
       "      <td>39.800827</td>\n",
       "      <td>202.847106</td>\n",
       "      <td>176.754750</td>\n",
       "      <td>269.179049</td>\n",
       "      <td>59.190718</td>\n",
       "      <td>...</td>\n",
       "      <td>52.599381</td>\n",
       "      <td>70.205312</td>\n",
       "      <td>80.329963</td>\n",
       "      <td>51.559799</td>\n",
       "      <td>124.787689</td>\n",
       "      <td>104.828364</td>\n",
       "      <td>38.132320</td>\n",
       "      <td>75.026077</td>\n",
       "      <td>37.963569</td>\n",
       "      <td>37.513253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_996122_1</th>\n",
       "      <td>3.959178</td>\n",
       "      <td>1.678095</td>\n",
       "      <td>1.323174</td>\n",
       "      <td>7.498730</td>\n",
       "      <td>21.946004</td>\n",
       "      <td>3.967258</td>\n",
       "      <td>43.177695</td>\n",
       "      <td>238.240213</td>\n",
       "      <td>1.691548</td>\n",
       "      <td>2.118184</td>\n",
       "      <td>...</td>\n",
       "      <td>2.291872</td>\n",
       "      <td>3.622075</td>\n",
       "      <td>567.383281</td>\n",
       "      <td>1.372061</td>\n",
       "      <td>85.329175</td>\n",
       "      <td>78.468709</td>\n",
       "      <td>20.547345</td>\n",
       "      <td>17.674589</td>\n",
       "      <td>11.628088</td>\n",
       "      <td>13.298897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_997245_1</th>\n",
       "      <td>8.710427</td>\n",
       "      <td>5.835441</td>\n",
       "      <td>8.072043</td>\n",
       "      <td>13.406415</td>\n",
       "      <td>62.608661</td>\n",
       "      <td>6.913223</td>\n",
       "      <td>21.131240</td>\n",
       "      <td>334.887067</td>\n",
       "      <td>10.083635</td>\n",
       "      <td>9.986821</td>\n",
       "      <td>...</td>\n",
       "      <td>13.184112</td>\n",
       "      <td>320.948201</td>\n",
       "      <td>18.947015</td>\n",
       "      <td>3.636536</td>\n",
       "      <td>92.492587</td>\n",
       "      <td>37.827363</td>\n",
       "      <td>4.753547</td>\n",
       "      <td>11.551204</td>\n",
       "      <td>4.308847</td>\n",
       "      <td>29.672485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_997813_2</th>\n",
       "      <td>19.900002</td>\n",
       "      <td>115.456455</td>\n",
       "      <td>102.160439</td>\n",
       "      <td>73.110311</td>\n",
       "      <td>55.923188</td>\n",
       "      <td>38.617480</td>\n",
       "      <td>38.744904</td>\n",
       "      <td>138.694299</td>\n",
       "      <td>80.817025</td>\n",
       "      <td>20.941561</td>\n",
       "      <td>...</td>\n",
       "      <td>142.779126</td>\n",
       "      <td>2.436368</td>\n",
       "      <td>92.180732</td>\n",
       "      <td>45.741706</td>\n",
       "      <td>27.204698</td>\n",
       "      <td>108.503469</td>\n",
       "      <td>67.252983</td>\n",
       "      <td>144.743062</td>\n",
       "      <td>102.948664</td>\n",
       "      <td>78.854286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_997813_3</th>\n",
       "      <td>10.997851</td>\n",
       "      <td>158.856644</td>\n",
       "      <td>90.649466</td>\n",
       "      <td>21.047259</td>\n",
       "      <td>109.146743</td>\n",
       "      <td>19.985383</td>\n",
       "      <td>18.103251</td>\n",
       "      <td>52.708553</td>\n",
       "      <td>39.793678</td>\n",
       "      <td>31.804228</td>\n",
       "      <td>...</td>\n",
       "      <td>93.152357</td>\n",
       "      <td>29.941569</td>\n",
       "      <td>207.961046</td>\n",
       "      <td>63.109370</td>\n",
       "      <td>42.633071</td>\n",
       "      <td>117.228353</td>\n",
       "      <td>64.383382</td>\n",
       "      <td>386.126155</td>\n",
       "      <td>95.920584</td>\n",
       "      <td>72.614930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>k99_99970_1</th>\n",
       "      <td>47.707257</td>\n",
       "      <td>100.118122</td>\n",
       "      <td>120.992256</td>\n",
       "      <td>28.672035</td>\n",
       "      <td>123.491480</td>\n",
       "      <td>45.090449</td>\n",
       "      <td>21.387333</td>\n",
       "      <td>97.332658</td>\n",
       "      <td>84.433542</td>\n",
       "      <td>62.712090</td>\n",
       "      <td>...</td>\n",
       "      <td>66.542855</td>\n",
       "      <td>297.587182</td>\n",
       "      <td>85.905713</td>\n",
       "      <td>92.736934</td>\n",
       "      <td>27.742234</td>\n",
       "      <td>27.638691</td>\n",
       "      <td>104.012192</td>\n",
       "      <td>9.661892</td>\n",
       "      <td>94.329046</td>\n",
       "      <td>40.535102</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10133 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           H16          H3          L4         L11  \\\n",
       "datarest$ProteinID                                                   \n",
       "k99_1000928_1       111.069493   94.231149  166.363367   99.088192   \n",
       "k99_1001938_3        46.630938  320.607204  264.045104   74.501569   \n",
       "k99_1001990_2        57.391732   47.461439  123.003403   32.235909   \n",
       "k99_1002150_3        30.935689  125.472858  347.763489  148.560410   \n",
       "k99_1002370_6        46.583836  164.440568  166.747069  134.089458   \n",
       "k99_1002967_3        27.341552  120.232326  349.095667  193.395062   \n",
       "k99_1003965_10       24.376980  172.249152  150.505049   18.110006   \n",
       "k99_1004041_1        53.284447    0.306497    4.638289   61.191568   \n",
       "k99_1004623_1       119.368003   33.717070   12.766574   38.006418   \n",
       "k99_1005212_10       54.244065  267.630639   62.279420  124.831251   \n",
       "k99_1005337_1       154.392649   83.607374  272.072922   63.092778   \n",
       "k99_1005475_1        41.334795   52.794187  132.989227   34.285508   \n",
       "k99_1005733_1       131.967098   23.004395  116.745592   46.300057   \n",
       "k99_1006916_3       175.810806    0.139214    5.430354   25.455806   \n",
       "k99_1006998_1       217.632768   32.915850  277.607949   32.080584   \n",
       "k99_1007311_1       172.994521  119.471056    5.150463   20.349668   \n",
       "k99_1007631_1        36.864080  117.667665  216.172100  126.561557   \n",
       "k99_1007659_2       323.951407    2.216117   11.720689   30.269739   \n",
       "k99_100772_1        221.756668  507.788938    5.102968  103.101741   \n",
       "k99_1007949_1        62.968256   58.710044  281.423659   22.513453   \n",
       "k99_1008105_1        97.160518   53.102530   46.897418  140.513848   \n",
       "k99_1008105_2        72.357921   20.590128   19.745163   43.018586   \n",
       "k99_1008105_3       233.355649   74.324369   15.008242   27.759816   \n",
       "k99_1008180_1        15.187049  113.536625   13.021212   61.337747   \n",
       "k99_1008218_1        56.584412  138.779130  242.561476   15.886922   \n",
       "k99_1008435_3        36.968736  173.767161   95.189930  127.752392   \n",
       "k99_1008548_2       114.743947   44.819853   65.549676   24.236662   \n",
       "k99_1008597_2       336.860031   10.450378   12.066680   51.704662   \n",
       "k99_1008705_2        83.383882   53.365558   47.799088   58.974744   \n",
       "k99_1008991_1        14.183006   21.144467   16.343396   54.560932   \n",
       "...                        ...         ...         ...         ...   \n",
       "k99_986881_2         71.725067   46.142810  249.027866   20.509308   \n",
       "k99_988185_3        284.722747  109.255008   21.199096  113.321378   \n",
       "k99_989559_1         53.745577  285.744149  109.222848   30.145157   \n",
       "k99_989607_1         21.372486    5.238690    4.239787    7.797743   \n",
       "k99_989714_5         18.102763   33.725137   78.541373   64.782197   \n",
       "k99_989874_5        436.717957    7.261263   35.282143   97.395999   \n",
       "k99_990427_1         88.757610   70.961000  151.333799   36.360445   \n",
       "k99_990682_2          3.124727   76.672831  102.157108   59.000290   \n",
       "k99_991061_1         36.536070   25.212403  200.306933   25.807963   \n",
       "k99_991419_1          3.012734    0.592461    2.226710   23.017513   \n",
       "k99_991741_1         36.078577    6.200558   11.098007   16.391302   \n",
       "k99_991826_1          8.093804   52.305888  117.718034   58.529716   \n",
       "k99_991826_3         19.348388   15.955913   54.466123   38.363492   \n",
       "k99_99188_2          40.975287   35.414517   17.301511   92.112643   \n",
       "k99_992336_2         69.489179  110.334996   47.178478    0.706841   \n",
       "k99_992839_2         41.191678  236.993691  582.003353   26.288025   \n",
       "k99_99300_10         57.533476   51.688237   70.426056  218.373047   \n",
       "k99_993197_4          7.388702   18.812668    3.064288   11.812190   \n",
       "k99_993759_8         72.136854   45.623568   39.880100   42.682101   \n",
       "k99_9941_1          421.955852   70.549217   30.678601  118.981610   \n",
       "k99_99441_1          41.289352   93.724856   64.446688  300.902263   \n",
       "k99_99524_2         778.730579   11.520877   19.098173   40.797132   \n",
       "k99_995394_1         88.744787    6.591762  184.875953   81.745859   \n",
       "k99_995896_4        141.243431  107.025153  141.962239   20.507947   \n",
       "k99_996093_4         41.399251   64.807983   61.345281   80.127394   \n",
       "k99_996122_1          3.959178    1.678095    1.323174    7.498730   \n",
       "k99_997245_1          8.710427    5.835441    8.072043   13.406415   \n",
       "k99_997813_2         19.900002  115.456455  102.160439   73.110311   \n",
       "k99_997813_3         10.997851  158.856644   90.649466   21.047259   \n",
       "k99_99970_1          47.707257  100.118122  120.992256   28.672035   \n",
       "\n",
       "                             H8          L3         H11         H20  \\\n",
       "datarest$ProteinID                                                    \n",
       "k99_1000928_1         54.591755  173.691705   50.662214  134.574253   \n",
       "k99_1001938_3         33.129914  149.089471  239.364036   60.129463   \n",
       "k99_1001990_2         66.112950   69.628543  106.518997  705.535557   \n",
       "k99_1002150_3         39.245119   43.271356  129.515300   79.409804   \n",
       "k99_1002370_6         14.935078   52.832114   52.968206  178.658634   \n",
       "k99_1002967_3         48.804329   53.359054   47.009773  152.000113   \n",
       "k99_1003965_10       132.123840   28.703120    8.149837  115.727792   \n",
       "k99_1004041_1        203.231975   88.284329    3.046813    5.299889   \n",
       "k99_1004623_1        117.899451  185.485691  446.934636   11.686298   \n",
       "k99_1005212_10        56.083609   54.110338   70.235402  175.656809   \n",
       "k99_1005337_1         32.272982   24.834490   86.632449  394.377900   \n",
       "k99_1005475_1         54.319255   87.355216   26.796499  131.341136   \n",
       "k99_1005733_1         69.666192   45.045068   55.899728    2.445917   \n",
       "k99_1006916_3         46.773192  102.757996   19.256817    3.847794   \n",
       "k99_1006998_1        108.743595   71.300360   38.116813  100.084247   \n",
       "k99_1007311_1         69.172643    3.112321   29.750006    7.817663   \n",
       "k99_1007631_1         29.536866  148.119197  106.115331  173.137766   \n",
       "k99_1007659_2          3.051028  530.322777   41.138164   10.867847   \n",
       "k99_100772_1         195.245184   24.812102   62.937204    5.882766   \n",
       "k99_1007949_1         51.175672  125.605334   37.996999  264.320399   \n",
       "k99_1008105_1         35.394027  120.324332  244.362456  602.312953   \n",
       "k99_1008105_2         39.804201   79.228430   94.875345  747.386813   \n",
       "k99_1008105_3         68.564973   74.368965   43.459606  353.385122   \n",
       "k99_1008180_1        102.489074   26.471033  223.678836   39.319820   \n",
       "k99_1008218_1         69.635360    9.417897   55.037163   54.076108   \n",
       "k99_1008435_3         24.688385  187.116384   40.011542  101.439969   \n",
       "k99_1008548_2        103.924199  105.443975   25.129432   34.657164   \n",
       "k99_1008597_2         66.516863  121.469664  109.961704   83.462251   \n",
       "k99_1008705_2         92.913103   61.585567  161.752034   24.575543   \n",
       "k99_1008991_1         27.145712  248.087044   20.066131   10.252830   \n",
       "...                         ...         ...         ...         ...   \n",
       "k99_986881_2          63.446196   63.527754   24.583542   96.667323   \n",
       "k99_988185_3         118.419744   80.363294   42.946036  464.331237   \n",
       "k99_989559_1          59.419067  101.868692   91.933955  199.032317   \n",
       "k99_989607_1          17.208427   84.646844   10.535081   57.181267   \n",
       "k99_989714_5          25.561647  143.368810   28.542558   24.711359   \n",
       "k99_989874_5          39.009174  248.418095   40.658572   23.025467   \n",
       "k99_990427_1         140.097476    6.625953   45.700091   58.809509   \n",
       "k99_990682_2          30.144200   82.896231   10.458825   13.273222   \n",
       "k99_991061_1          47.756136   57.971996  269.991030  129.861009   \n",
       "k99_991419_1           3.387003    1.712442   22.779307    1.775004   \n",
       "k99_991741_1           6.033534    6.282880   29.432023    1.379454   \n",
       "k99_991826_1          47.984053   85.682910   12.752715   13.151539   \n",
       "k99_991826_3          23.794290   51.280761   54.685985   13.570566   \n",
       "k99_99188_2          411.046092   59.960873   84.538943  114.003832   \n",
       "k99_992336_2          53.415820  110.156962  181.152375   82.686340   \n",
       "k99_992839_2          83.354495  582.233477   87.958774   83.684066   \n",
       "k99_99300_10        1001.555339   63.822298   52.255680   60.389625   \n",
       "k99_993197_4          16.704778   44.678333   30.332263    1.236302   \n",
       "k99_993759_8          40.751973   19.616386   98.518508  344.930759   \n",
       "k99_9941_1            70.286791   69.750482  145.949478  244.268993   \n",
       "k99_99441_1         1564.878900   11.398853  119.783493   34.885255   \n",
       "k99_99524_2          407.020702    2.038803   72.388909   11.173429   \n",
       "k99_995394_1          46.235319  103.827076   67.934138   29.360562   \n",
       "k99_995896_4         112.438742   13.333790  141.704716   42.099626   \n",
       "k99_996093_4         117.044491   39.800827  202.847106  176.754750   \n",
       "k99_996122_1          21.946004    3.967258   43.177695  238.240213   \n",
       "k99_997245_1          62.608661    6.913223   21.131240  334.887067   \n",
       "k99_997813_2          55.923188   38.617480   38.744904  138.694299   \n",
       "k99_997813_3         109.146743   19.985383   18.103251   52.708553   \n",
       "k99_99970_1          123.491480   45.090449   21.387333   97.332658   \n",
       "\n",
       "                           L10          L7     ...             L16  \\\n",
       "datarest$ProteinID                             ...                   \n",
       "k99_1000928_1        26.011840   59.754656     ...       68.093307   \n",
       "k99_1001938_3         6.352032   16.593794     ...       88.449492   \n",
       "k99_1001990_2       116.762120  146.875607     ...       17.494716   \n",
       "k99_1002150_3        25.220030   71.962896     ...       62.935518   \n",
       "k99_1002370_6        62.721379   55.951204     ...      162.402114   \n",
       "k99_1002967_3        58.934078   43.200448     ...       87.707173   \n",
       "k99_1003965_10       21.424122   16.858031     ...       57.223006   \n",
       "k99_1004041_1       195.578089    6.252366     ...      130.641019   \n",
       "k99_1004623_1       182.239036   24.190626     ...       33.647304   \n",
       "k99_1005212_10       51.662677   47.703592     ...       95.664787   \n",
       "k99_1005337_1        16.996699   14.575021     ...      143.982645   \n",
       "k99_1005475_1        50.314613  108.054515     ...       83.264222   \n",
       "k99_1005733_1        17.834656   61.272581     ...       97.824029   \n",
       "k99_1006916_3       144.468301  113.894306     ...      177.030833   \n",
       "k99_1006998_1        75.199448   98.662185     ...      112.682155   \n",
       "k99_1007311_1        87.060368  219.169475     ...       83.179488   \n",
       "k99_1007631_1        10.344472   68.278372     ...       26.826830   \n",
       "k99_1007659_2        38.951188   18.246542     ...      187.940755   \n",
       "k99_100772_1        161.211253   73.688159     ...       95.280290   \n",
       "k99_1007949_1        33.884970  116.922834     ...       79.150747   \n",
       "k99_1008105_1        68.507910   10.580655     ...       48.559364   \n",
       "k99_1008105_2        60.187154    7.833829     ...       43.691513   \n",
       "k99_1008105_3        33.754965    7.039364     ...       31.286745   \n",
       "k99_1008180_1        47.393270   94.291313     ...       73.628504   \n",
       "k99_1008218_1        31.608293   22.455110     ...      121.542235   \n",
       "k99_1008435_3        23.529278   19.802737     ...       37.230945   \n",
       "k99_1008548_2        96.906225   24.770467     ...       42.855305   \n",
       "k99_1008597_2         3.592345   89.744451     ...      262.603783   \n",
       "k99_1008705_2        97.624925  114.127828     ...       85.751608   \n",
       "k99_1008991_1       239.079743  482.507133     ...        4.575419   \n",
       "...                        ...         ...     ...             ...   \n",
       "k99_986881_2         52.329958   64.337782     ...       64.452276   \n",
       "k99_988185_3         49.561183    6.171331     ...       23.416377   \n",
       "k99_989559_1        124.424764   75.420037     ...      191.374727   \n",
       "k99_989607_1          4.448012   40.092313     ...        6.967357   \n",
       "k99_989714_5        171.413633  280.820581     ...       41.318265   \n",
       "k99_989874_5        191.905049    6.042956     ...       79.741161   \n",
       "k99_990427_1         40.696426    7.899579     ...      110.116921   \n",
       "k99_990682_2        125.100109    0.376024     ...       83.164092   \n",
       "k99_991061_1         41.787830   72.995633     ...       63.342419   \n",
       "k99_991419_1        333.171941  182.074637     ...       13.904292   \n",
       "k99_991741_1         12.175121  756.155606     ...       24.618277   \n",
       "k99_991826_1         75.009943   27.793513     ...       48.497533   \n",
       "k99_991826_3        169.254131  187.965419     ...       45.318710   \n",
       "k99_99188_2          25.825147   69.377144     ...       60.121992   \n",
       "k99_992336_2         52.833980  116.850245     ...       81.585943   \n",
       "k99_992839_2         71.018943   18.976264     ...      146.727643   \n",
       "k99_99300_10        109.716554  145.773820     ...      102.388925   \n",
       "k99_993197_4         30.071025   91.500718     ...       53.954294   \n",
       "k99_993759_8         85.579857   39.813719     ...        4.379129   \n",
       "k99_9941_1           21.943717   14.361836     ...       20.626701   \n",
       "k99_99441_1           3.385661    3.022282     ...        1.648258   \n",
       "k99_99524_2          15.710797    6.437326     ...        3.160433   \n",
       "k99_995394_1         82.055675  132.537694     ...      141.450007   \n",
       "k99_995896_4         71.641031   76.769254     ...       62.294905   \n",
       "k99_996093_4        269.179049   59.190718     ...       52.599381   \n",
       "k99_996122_1          1.691548    2.118184     ...        2.291872   \n",
       "k99_997245_1         10.083635    9.986821     ...       13.184112   \n",
       "k99_997813_2         80.817025   20.941561     ...      142.779126   \n",
       "k99_997813_3         39.793678   31.804228     ...       93.152357   \n",
       "k99_99970_1          84.433542   62.712090     ...       66.542855   \n",
       "\n",
       "                            H14           H1          L1          H6  \\\n",
       "datarest$ProteinID                                                     \n",
       "k99_1000928_1         71.056933    33.271179  161.770360   70.540445   \n",
       "k99_1001938_3        410.993816    15.131730   19.084435    3.638069   \n",
       "k99_1001990_2         12.224755    34.637102   12.661787  126.771758   \n",
       "k99_1002150_3         21.794327    36.540982   79.051493   42.634040   \n",
       "k99_1002370_6          1.037451   144.370090  208.113864   14.262005   \n",
       "k99_1002967_3          9.256106    59.617181  247.086797   24.663200   \n",
       "k99_1003965_10        41.482728   204.751568   35.337095   53.272303   \n",
       "k99_1004041_1          9.193953     4.496304  142.587605    0.826234   \n",
       "k99_1004623_1        118.834157    46.292226   35.476515   33.872342   \n",
       "k99_1005212_10         7.083284   126.750742  115.386956   37.891127   \n",
       "k99_1005337_1          1.291334    13.995447   84.466980   17.595855   \n",
       "k99_1005475_1         81.462515    42.030277  165.868790   53.777821   \n",
       "k99_1005733_1         10.591840   104.433287  103.352557   69.095019   \n",
       "k99_1006916_3         10.818418    53.110737  106.636277  106.527928   \n",
       "k99_1006998_1        341.052887    62.696106   14.234573   65.602727   \n",
       "k99_1007311_1        287.281548    66.359367  156.395535   43.959460   \n",
       "k99_1007631_1         61.329672    65.778732   24.884930    7.335209   \n",
       "k99_1007659_2         22.869324    14.214832  134.948844  327.872525   \n",
       "k99_100772_1          34.031023   139.726011   58.911694   72.156834   \n",
       "k99_1007949_1          2.793514    50.310957  222.896789   85.199731   \n",
       "k99_1008105_1         58.858740    57.157386   11.490327   42.118520   \n",
       "k99_1008105_2         51.254832    64.158146   20.654935   42.772805   \n",
       "k99_1008105_3         33.925599    39.628948   22.844824   41.541536   \n",
       "k99_1008180_1        291.178040    50.062576   27.931189   56.354810   \n",
       "k99_1008218_1         16.536745    22.638705  138.888164   64.707997   \n",
       "k99_1008435_3        264.905214    40.525079  116.844388   37.470565   \n",
       "k99_1008548_2        150.394764    63.493967  184.482410   42.900626   \n",
       "k99_1008597_2          0.864858   175.382372   25.034425   71.264266   \n",
       "k99_1008705_2        132.415589    66.312452  120.764616   95.294835   \n",
       "k99_1008991_1          2.029517    24.949074   22.483890  275.678256   \n",
       "...                         ...          ...         ...         ...   \n",
       "k99_986881_2         158.935581    65.364268  152.094888   47.966426   \n",
       "k99_988185_3          54.173183    37.626349  106.743917   38.127006   \n",
       "k99_989559_1          27.189949    52.427279   87.246854   48.851018   \n",
       "k99_989607_1        1243.113262    22.678447  541.383898   38.994140   \n",
       "k99_989714_5         173.799895    46.810370  229.354717  138.120506   \n",
       "k99_989874_5          19.548117    89.645424   18.464332  431.256857   \n",
       "k99_990427_1          10.819953    34.073402   13.432432   47.333795   \n",
       "k99_990682_2         111.125592    30.983864   24.564292  159.006282   \n",
       "k99_991061_1          94.161809    51.982227  326.181419   37.302418   \n",
       "k99_991419_1           6.770838    19.667648  113.029613  102.713870   \n",
       "k99_991741_1          24.585423    78.446327  161.702029   28.955127   \n",
       "k99_991826_1         178.090202    58.703498   17.100617   44.996199   \n",
       "k99_991826_3         190.781121   140.921684  580.472026   15.913658   \n",
       "k99_99188_2          241.589395    43.355378  112.401977   43.606543   \n",
       "k99_992336_2           0.860196    68.124300   15.599012   49.605158   \n",
       "k99_992839_2          66.717530    56.427479   28.374838  344.149848   \n",
       "k99_99300_10         284.636193    31.342840  146.327508   34.910865   \n",
       "k99_993197_4          13.450632    14.947777    8.502076   32.254392   \n",
       "k99_993759_8           4.373121   149.948053   53.937557  208.164519   \n",
       "k99_9941_1            62.686112    34.038831   64.669907   73.436938   \n",
       "k99_99441_1            1.586085  1338.174649    6.760904   28.920598   \n",
       "k99_99524_2            5.150197    24.043758  120.281227   34.992531   \n",
       "k99_995394_1         199.246303    74.208047  102.196755  104.131953   \n",
       "k99_995896_4         174.828352   145.722862   35.921944   44.932540   \n",
       "k99_996093_4          70.205312    80.329963   51.559799  124.787689   \n",
       "k99_996122_1           3.622075   567.383281    1.372061   85.329175   \n",
       "k99_997245_1         320.948201    18.947015    3.636536   92.492587   \n",
       "k99_997813_2           2.436368    92.180732   45.741706   27.204698   \n",
       "k99_997813_3          29.941569   207.961046   63.109370   42.633071   \n",
       "k99_99970_1          297.587182    85.905713   92.736934   27.742234   \n",
       "\n",
       "                           L14          H2          H10         L17  \\\n",
       "datarest$ProteinID                                                    \n",
       "k99_1000928_1       101.274311  149.492346    99.904085   58.012858   \n",
       "k99_1001938_3         2.979719  106.715396    22.169808   66.089185   \n",
       "k99_1001990_2       104.163969   37.907910   168.860490   30.236346   \n",
       "k99_1002150_3        61.234510   69.066572   725.394872  136.957015   \n",
       "k99_1002370_6        43.673788   74.932959    38.915743   45.162795   \n",
       "k99_1002967_3        51.850043    4.419552    27.218740   21.916938   \n",
       "k99_1003965_10      127.598116   74.795004    93.190566   92.367805   \n",
       "k99_1004041_1       266.520335    8.812933    52.492953   17.674539   \n",
       "k99_1004623_1        52.061963   36.413021   111.688583   22.764500   \n",
       "k99_1005212_10       82.778597   97.051830   151.377216  138.447806   \n",
       "k99_1005337_1        26.110809   72.402559    19.235825   45.623446   \n",
       "k99_1005475_1        67.575000  134.316490    44.658326   73.540616   \n",
       "k99_1005733_1        48.348803  475.281984    13.203118  111.412930   \n",
       "k99_1006916_3        50.580339  269.741368    20.338973   18.281009   \n",
       "k99_1006998_1       137.541238   16.956252   105.403932   17.336581   \n",
       "k99_1007311_1        52.131438  228.362410    26.485740  196.476485   \n",
       "k99_1007631_1         9.038559   46.606115  1128.870371   15.314261   \n",
       "k99_1007659_2        62.831065  302.385864    24.794732    7.176967   \n",
       "k99_100772_1         48.211895  373.702282    23.264329  121.162355   \n",
       "k99_1007949_1        49.285397  178.029218    31.586833   68.288730   \n",
       "k99_1008105_1        23.934693   35.413779    30.884223  153.488932   \n",
       "k99_1008105_2        19.853697   49.171634   345.129520  155.586749   \n",
       "k99_1008105_3        22.624244   51.710949    28.625361  164.788802   \n",
       "k99_1008180_1        26.735054  235.855610   111.598174   28.519930   \n",
       "k99_1008218_1        57.914429  171.654608    36.243821   40.396772   \n",
       "k99_1008435_3        51.387261  140.366705    61.901285   91.628428   \n",
       "k99_1008548_2       122.452632  195.462071    47.728523  131.932277   \n",
       "k99_1008597_2        55.105001  606.904975     7.139729  193.304802   \n",
       "k99_1008705_2       185.325284  127.312305    71.008522   72.224352   \n",
       "k99_1008991_1        45.283593   43.438743    24.628750    7.154000   \n",
       "...                        ...         ...          ...         ...   \n",
       "k99_986881_2         60.565922   94.545120    28.180323  121.394564   \n",
       "k99_988185_3         65.221674   50.037052    25.940555  139.556174   \n",
       "k99_989559_1         64.941353   33.978277    12.991661   62.919910   \n",
       "k99_989607_1         39.977930  163.316391    53.464826   84.602661   \n",
       "k99_989714_5        181.006556   99.057888    58.995529   82.048609   \n",
       "k99_989874_5         54.006182   28.161478    48.214084   59.162636   \n",
       "k99_990427_1        149.829822  327.653439    62.410996   14.494765   \n",
       "k99_990682_2         30.446200  121.973523   411.428478  163.663505   \n",
       "k99_991061_1         38.456098  146.162365    15.851840   80.470856   \n",
       "k99_991419_1        187.718389  354.255053    84.073836   11.267959   \n",
       "k99_991741_1         96.863843    6.997331    12.687838   28.004871   \n",
       "k99_991826_1         28.637388  105.409735   400.144006  258.115744   \n",
       "k99_991826_3         23.240362   32.949575   130.358814   59.896302   \n",
       "k99_99188_2          42.533944  101.070831    38.945893  258.366007   \n",
       "k99_992336_2        164.542825   30.371297   525.937837   35.865205   \n",
       "k99_992839_2         52.515811  236.883788   130.909592  123.933795   \n",
       "k99_99300_10         62.307465  128.704373    53.509529  110.656657   \n",
       "k99_993197_4         18.588861   13.478309   615.029349  337.343961   \n",
       "k99_993759_8         97.324389    7.107926    53.071577    2.960770   \n",
       "k99_9941_1           77.250724  155.156430    27.444070  118.640369   \n",
       "k99_99441_1           6.678438   14.575007    50.024170    7.771759   \n",
       "k99_99524_2          96.598168   66.783737    91.903437  114.752923   \n",
       "k99_995394_1        116.885412   39.239192    66.832456   99.410804   \n",
       "k99_995896_4        193.934256   26.622678    38.868802   65.478220   \n",
       "k99_996093_4        104.828364   38.132320    75.026077   37.963569   \n",
       "k99_996122_1         78.468709   20.547345    17.674589   11.628088   \n",
       "k99_997245_1         37.827363    4.753547    11.551204    4.308847   \n",
       "k99_997813_2        108.503469   67.252983   144.743062  102.948664   \n",
       "k99_997813_3        117.228353   64.383382   386.126155   95.920584   \n",
       "k99_99970_1          27.638691  104.012192     9.661892   94.329046   \n",
       "\n",
       "                           L26  \n",
       "datarest$ProteinID              \n",
       "k99_1000928_1        60.128184  \n",
       "k99_1001938_3        35.278634  \n",
       "k99_1001990_2       143.500305  \n",
       "k99_1002150_3        48.171145  \n",
       "k99_1002370_6        79.388883  \n",
       "k99_1002967_3        68.751905  \n",
       "k99_1003965_10      104.356005  \n",
       "k99_1004041_1        40.778811  \n",
       "k99_1004623_1        39.151955  \n",
       "k99_1005212_10       85.003125  \n",
       "k99_1005337_1        87.443748  \n",
       "k99_1005475_1        53.925726  \n",
       "k99_1005733_1       141.809785  \n",
       "k99_1006916_3        82.846367  \n",
       "k99_1006998_1        22.917993  \n",
       "k99_1007311_1        78.682065  \n",
       "k99_1007631_1         9.883403  \n",
       "k99_1007659_2       193.394550  \n",
       "k99_100772_1        335.377411  \n",
       "k99_1007949_1        83.179259  \n",
       "k99_1008105_1        30.721960  \n",
       "k99_1008105_2        30.769563  \n",
       "k99_1008105_3        13.913015  \n",
       "k99_1008180_1        45.642919  \n",
       "k99_1008218_1       119.941563  \n",
       "k99_1008435_3        44.829909  \n",
       "k99_1008548_2        58.422157  \n",
       "k99_1008597_2        42.680944  \n",
       "k99_1008705_2        65.413707  \n",
       "k99_1008991_1       285.272588  \n",
       "...                        ...  \n",
       "k99_986881_2         60.364492  \n",
       "k99_988185_3         16.241541  \n",
       "k99_989559_1         11.676106  \n",
       "k99_989607_1         16.568652  \n",
       "k99_989714_5        144.450007  \n",
       "k99_989874_5         15.641727  \n",
       "k99_990427_1         90.445981  \n",
       "k99_990682_2         56.555794  \n",
       "k99_991061_1         57.616524  \n",
       "k99_991419_1         11.200960  \n",
       "k99_991741_1         72.402798  \n",
       "k99_991826_1         49.954926  \n",
       "k99_991826_3         21.846505  \n",
       "k99_99188_2          24.085724  \n",
       "k99_992336_2         66.438763  \n",
       "k99_992839_2        102.075627  \n",
       "k99_99300_10         47.255105  \n",
       "k99_993197_4         16.132682  \n",
       "k99_993759_8          6.403968  \n",
       "k99_9941_1           34.738022  \n",
       "k99_99441_1           8.543831  \n",
       "k99_99524_2          16.189056  \n",
       "k99_995394_1        122.105325  \n",
       "k99_995896_4        138.665934  \n",
       "k99_996093_4         37.513253  \n",
       "k99_996122_1         13.298897  \n",
       "k99_997245_1         29.672485  \n",
       "k99_997813_2         78.854286  \n",
       "k99_997813_3         72.614930  \n",
       "k99_99970_1          40.535102  \n",
       "\n",
       "[10133 rows x 40 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Remove any human derived proteins\n",
    "df = df[df.index.str.contains('k99_') != False]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df.to_csv('./2Search/CSVs/NormalizedCommonReps_nohuman.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create Biom Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#2 Search pDB Approach - no human\n",
    "#Convert tab-separated file to biom file\n",
    "!biom convert -i ./2Search/CSVs/NormalizedCommonReps_nohuman.txt \\\n",
    "-o ./2Search/NormalizedCommonReps2_nohuman.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#2 Search pDB Approach\n",
    "#Convert tab-separated file to biom file\n",
    "!biom convert -i ./2Search/CSVs/NormalizedCommonReps.txt \\\n",
    "-o ./2Search/NormalizedCommonReps2.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Metabolomics\n",
    "#Convert tab-separated file to biom file\n",
    "!biom convert -i ../Metabolomics/Feature_table.txt \\\n",
    "-o ./Metabolomics.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#16S\n",
    "#Convert tab-separated file to biom file\n",
    "!biom convert -i ../Genomics/16S/reference-hit_idswap2_blankremove.txt \\\n",
    "-o ./16S.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Serum\n",
    "!biom convert -i ../Serum/CSVs/NormalizedCommonReps_ids.txt \\\n",
    "-o ./Serum_Common.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#MG\n",
    "!biom convert -i ../Genomics/Shotgun/Salmon_CPMs_0s.txt \\\n",
    "-o ./Salmon_CPMs.biom \\\n",
    "-m ../UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import all as Qiime2 artifacts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./Metabolomics.biom \\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path Metabolomics_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./2Search/NormalizedCommonReps2.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path 2search_common_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./16S.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path 16S_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./Salmon_CPMs.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path ./Salmon_CPMs_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./Serum_Common.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path ./Serum_Common_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path ./2Search/NormalizedCommonReps2_nohuman.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path ./pDB_Common_nohuman_biom.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Metagenomics using UniFrac. Based on centrifuge counts.\n",
    "!qiime tools import \\\n",
    "  --input-path ./Genomics/Shotgun/all.filt.biom \\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path MG_all_biom.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Feature table summarize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ./pDB_Common_nohuman_biom.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table ./pDB_Common_nohuman_biom.qza \\\n",
    "  --o-visualization ./pDB_Common_nohuman_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Metabolomics_biom.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table Metabolomics_biom.qza \\\n",
    "  --o-visualization Metabolomics_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: 2search_biom.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table 2search_common_biom.qza \\\n",
    "  --o-visualization 2search_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: 16S_biom.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table 16S_biom.qza \\\n",
    "  --o-visualization 16S_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ./Serum_Common_biom.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table ./Serum_Common_biom.qza \\\n",
    "  --o-visualization ./Serum_Common_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\r\n",
      "\r\n",
      "Aborted!\r\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table ./Salmon_CPMs_biom.qza \\\n",
    "  --o-visualization Salmon_CPMs_biom.qzv \\\n",
    "  --m-sample-metadata-file ../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "qiime feature-table summarize \\\n",
    "  --i-table ./Serum_biom.qza \\\n",
    "  --o-visualization Serum_biom.qzv \\\n",
    "  --m-sample-metadata-file ../../UC_MP_Emperor_Map.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Qiime 1 PCoAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Errors and/or warnings detected in mapping file.  Please check the log and html file for details.\r\n"
     ]
    }
   ],
   "source": [
    "!validate_mapping_file.py -o vmf-map/ -m ../UC_MP_Emperor_Map_10052018.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "vmf-map/<br>\n",
       "&nbsp;&nbsp;<a href='vmf-map/UC_MP_Emperor_Map_10052018.html' target='_blank'>UC_MP_Emperor_Map_10052018.html</a><br>\n",
       "&nbsp;&nbsp;<a href='vmf-map/overlib.js' target='_blank'>overlib.js</a><br>\n",
       "&nbsp;&nbsp;<a href='vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt' target='_blank'>UC_MP_Emperor_Map_10052018_corrected.txt</a><br>\n",
       "&nbsp;&nbsp;<a href='vmf-map/UC_MP_Emperor_Map_10052018.log' target='_blank'>UC_MP_Emperor_Map_10052018.log</a><br>"
      ],
      "text/plain": [
       "vmf-map/\n",
       "  UC_MP_Emperor_Map_10052018.html\n",
       "  overlib.js\n",
       "  UC_MP_Emperor_Map_10052018_corrected.txt\n",
       "  UC_MP_Emperor_Map_10052018.log"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import FileLinks, FileLink\n",
    "FileLinks('vmf-map/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -4.15747501546 and the largest is 2.36480772077e+16.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0035096125299 and the largest is 0.948397941533.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./Metabolomics.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_Metabolomics -p ../smallDB_correct/PCoA_Plots/paramaters2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./2Search/NormalizedCommonReps2.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_2search -p ../smallDB_correct/PCoA_Plots/paramaters2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0305262601337 and the largest is 1.70812132461.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0669606719016 and the largest is 3.17652373099.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./16S.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_16S -p ../smallDB_correct/PCoA_Plots/paramaters2_16S.txt -t ../Genomics/16S/insertion_tree.relabelled.tre "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./Salmon_CPMs.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_MG -p ../smallDB_correct/PCoA_Plots/paramaters2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.42588933238e-07 and the largest is 295304255.711.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00387462944193 and the largest is 0.497876640547.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./Serum_Common.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_Serum -p ../smallDB_correct/PCoA_Plots/paramaters2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.66231375683e-06 and the largest is 2297967245.68.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ./2Search/NormalizedCommonReps2_nohuman.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_2search_nohuman -p ../smallDB_correct/PCoA_Plots/paramaters2.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/bin/beta_diversity_through_plots.py\", line 4, in <module>\r\n",
      "    __import__('pkg_resources').run_script('qiime==1.9.1', 'beta_diversity_through_plots.py')\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 750, in run_script\r\n",
      "    self.require(requires)[0].run_script(script_name, ns)\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 1527, in run_script\r\n",
      "    exec(code, namespace, namespace)\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime-1.9.1-py2.7.egg-info/scripts/beta_diversity_through_plots.py\", line 153, in <module>\r\n",
      "    main()\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime-1.9.1-py2.7.egg-info/scripts/beta_diversity_through_plots.py\", line 127, in main\r\n",
      "    create_dir(output_dir, fail_on_exist=not opts.force)\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/util/_misc.py\", line 225, in create_dir\r\n",
      "    return ror(error_code_lookup['DIR_EXISTS'])\r\n",
      "  File \"/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/util/_misc.py\", line 334, in _handle_error_codes\r\n",
      "    raise OSError(error_strings[error_code])\r\n",
      "OSError: Directory already exists: ./Qiime/PCoA_Serum_nored\r\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ../Serum/UC_Serum_CommonReps_nored_hdf5.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_Serum_nored -p ../smallDB_correct/PCoA_Plots/paramaters2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:456: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00722017075537 and the largest is 1.82941390622e+13.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/cogent/maths/distance_transform.py:188: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  if rank(datamtx) != 2:\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0377227709131 and the largest is 2.30405445323.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00104850689555 and the largest is 1.30424230145.\n",
      "  RuntimeWarning\n",
      "/Users/rhmills/miniconda3/envs/qiime1/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.139427821991 and the largest is 8.31575816029.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "!beta_diversity_through_plots.py -i ../Genomics/Shotgun/all.filt.biom -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -o ./Qiime/PCoA_MG_allUniq -p ../smallDB_correct/PCoA_Plots/paramaters2_16S.txt -t ../Genomics/Shotgun/tree.nwk "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Qiime 1 Beta-Diversity Statistics Adonis & PERMANOVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/rhmills/Documents/Thesis Work/UC_Severity/pDB_Proteomics\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Data types which require PERMANOVA categorical significance.\n",
    "Perma = ['sex','race','historic_extent','ASA_exposure', 'current_5ASA', 'steroid_exposure', 'current_steroids', 'IM_exposure', 'IM_type', 'biologic_exposure',\n",
    "        'biologic_exposure_type','current_biologic','current_biologic_type','Experiment','TMT_Label']\n",
    "Adonis = ['CRP','Calprotectin', 'partial_mayo','age','age_diagnosis','disease_duration','height','stool_frequency','rectal_bleeding','PGA','mayo_endoscopic_score','UCEIS_endoscopic_score','COLLECTION_TIMESTAMP','Endoscopy_date']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on MG UniFrac\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_MG_allUniq/unweighted_unifrac_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/MG_UniFrac/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on MG UniFrac\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_MG_allUniq/unweighted_unifrac_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/MG_UniFrac/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on 16S\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_16S/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/16S_bray/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on 16S - all samples\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ../Genomics/16S/core-metrics-results_idswap2_newmetadata_allsamples/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/16S_bray/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on 16S\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_16S/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/16S_bray/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on pDB\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_2search/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/pDB_MP/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on pDB\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_2search/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/pDB_MP/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on pDB NOHUMAN\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_2search_nohuman/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/pDB_MP_nohuman/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on pDB NOHUMAN\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_2search_nohuman/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/pDB_MP_nohuman/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on Metabolomics\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_Metabolomics/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/Metabolomics/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on Metabolomics\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_Metabolomics/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/Metabolomics/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on Metagenome\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_MG/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/MG/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on Metagenome\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_MG/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/MG/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on Serum\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_Serum/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/Serum/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on Serum\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_Serum/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/Serum/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on Serum NORED\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_Serum_nored/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/Serum_nored/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on Serum NORED\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_Serum_nored/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/Serum_nored/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Permanovas on 16S\n",
    "for i in Perma:\n",
    "        !compare_categories.py --method permanova -i ./Qiime/PCoA_16S/unweighted_unifrac_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Permanova/16S/$i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Iterative Adonis on 16S\n",
    "for i in Adonis:\n",
    "        !compare_categories.py --method adonis -i ./Qiime/PCoA_16S/unweighted_unifrac_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c $i -o ./Qiime/Adonis/16S/$i -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!compare_categories.py --method permanova -i ./Qiime/PCoA_2search/bray_curtis_dm.txt -m ./vmf-map/UC_MP_Emperor_Map_10052018_corrected.txt -c IM_type -o ./Qiime/Permanova/pDB_MP/IM_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Core metrics - Qiime2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results_pDB_noHuman/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_pDB_noHuman/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_pDB_noHuman/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_pDB_noHuman/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_pDB_noHuman/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_pDB_noHuman/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_pDB_noHuman/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_pDB_noHuman/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_pDB_noHuman/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_pDB_noHuman/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#2Searh pDB using the no-human proteins\n",
    "!qiime diversity core-metrics \\\n",
    "  --i-table pDB_Common_nohuman_biom.qza \\\n",
    "    --p-sampling-depth 539826 \\\n",
    "--m-metadata-file ../UC_MP_Emperor_Map.txt \\\n",
    "--output-dir core-metrics-results_pDB_noHuman"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results_2search/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_2search/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_2search/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_2search/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_2search/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_2search/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_2search/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_2search/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_2search/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_2search/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Huge experimental (TMT experiment) variability in this approach.\n",
    "\n",
    "!qiime diversity core-metrics \\\n",
    "  --i-table 2search_common_biom.qza \\\n",
    "    --p-sampling-depth 694193 \\\n",
    "--m-metadata-file ../UC_MP_Emperor_Map.txt \\\n",
    "--output-dir core-metrics-results_2search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results_IGC/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_IGC/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_IGC/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_IGC/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_IGC/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_IGC/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_IGC/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_IGC/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_IGC/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_IGC/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Huge experimental (TMT experiment) variability in this approach.\n",
    "\n",
    "!qiime diversity core-metrics \\\n",
    "  --i-table IGC_biom.qza \\\n",
    "    --p-sampling-depth 1813092 \\\n",
    "--m-metadata-file ../UC_MP_Emperor_Map.txt \\\n",
    "--output-dir core-metrics-results_IGC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results_Metabolomics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_Metabolomics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_Metabolomics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results_Metabolomics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_Metabolomics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: core-metrics-results_Metabolomics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_Metabolomics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: core-metrics-results_Metabolomics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_Metabolomics/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: core-metrics-results_Metabolomics/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Metabolomics\n",
    "\n",
    "!qiime diversity core-metrics \\\n",
    "  --i-table Metabolomics_biom.qza \\\n",
    "    --p-sampling-depth 184074904 \\\n",
    "--m-metadata-file ../UC_MP_Emperor_Map.txt \\\n",
    "--output-dir core-metrics-results_Metabolomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Metabolome_shannon.qza\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "#Shannon diversity metabolome\n",
    "\n",
    "!qiime diversity alpha \\\n",
    "  --i-table Metabolomics_biom.qza \\\n",
    "  --p-metric shannon \\\n",
    "  --o-alpha-diversity Metabolome_shannon.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Random Forest Classifier - Qiime2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../Random_Forests/pieloue_rforest_16S.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "#Code for determining features most related to the pielou evenness - supplemental figure\n",
    "\n",
    "!qiime sample-classifier regress-samples \\\n",
    "--i-table ./16S_biom.qza \\\n",
    "--m-metadata-file ../UC_MP_Emperor_Map.txt \\\n",
    "--m-metadata-column pielou_e \\\n",
    "--p-optimize-feature-selection \\\n",
    "--p-parameter-tuning \\\n",
    "--p-estimator RandomForestRegressor \\\n",
    "--o-visualization \\\n",
    "../Random_Forests/pieloue_rforest_16S.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create combined table with all data\n",
    "Mb = pd.read_csv('./Metabolomics/Feature_table.txt', sep = '\\t', index_col = '#OTUID')\n",
    "Mp = pd.read_csv('./pDB_Proteomics/2Search/CSVs/NormalizedCommonReps.txt', sep = '\\t', index_col = 'datarest$ProteinID')\n",
    "Ser = pd.read_csv('./Serum/CSVs/NormalizedCommonReps_ids.txt', sep = '\\t', index_col = 'datarest$ProteinID')\n",
    "MG = pd.read_csv('./Genomics/Shotgun/Salmon_CPMs_0s.txt', sep = '\\t', index_col = '#OTU')\n",
    "Amp = pd.read_csv('./Genomics/16S/reference-hit_idswap2_noblanks.txt', sep = '\\t', index_col = '#OTU ID')\n",
    "\n",
    "#Overlapping IDS are a problem, so append the metaproteome ids with _MP to signify metaproteome.\n",
    "Mp.index = Mp.index + '_MP'\n",
    "\n",
    "#Concatenate data\n",
    "Objs = [Mb, Mp, Ser, MG, Amp]\n",
    "alldf = pd.concat(Objs)\n",
    "\n",
    "alldf.index.rename('Features', inplace = True)\n",
    "\n",
    "#Save the new feature table\n",
    "#alldf.to_csv('./allfeaturesconcat.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create biom file\n",
    "!biom convert -i ./allfeaturesconcat.txt \\\n",
    "-o ./Allfeaturesconcat.biom \\\n",
    "-m ./UC_MP_Emperor_Map_1.txt \\\n",
    "--table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<i> Performed random forest analyses on supercomputer using the Allfeaturesconcat.biom file </i>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 16S analysis - Qiime2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initializes the notebook with inline display\n",
    "%matplotlib inline\n",
    "\n",
    "from os import mkdir\n",
    "import os\n",
    "import copy\n",
    "from os.path import abspath, join as pjoin, exists\n",
    "from shutil import copy2, move\n",
    "from time import strftime, strptime\n",
    "from numpy import nan, isnan, arange\n",
    "from pandas import read_csv, Series, DataFrame\n",
    "from IPython.display import Image\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "from IPython.display import FileLinks, FileLink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!validate_mapping_file.py -o vmf-map/ -m ./11549_prep_4716_qiime_20180508-114955.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i ./reference-hit_idswap2.txt -o ./reference-hit_idswap2.biom --table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import all files as Qiime2 artifacts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path reference-hit_idswap2.biom \\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --output-path biom_id2.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime feature-table summarize \\\n",
    "  --i-table biom_id2.qza \\\n",
    "  --o-visualization biom_id2.qzv \\\n",
    "  --m-sample-metadata-file ./UC_Severity_1_MF_Idswap6.21.18.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## rep-seqs \n",
    "!qiime tools import \\\n",
    "  --input-path reference-hit.seqs.fa \\\n",
    "  --output-path sequences.qza \\\n",
    "  --type 'FeatureData[Sequence]'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime alignment mafft \\\n",
    "  --i-sequences sequences.qza \\\n",
    "  --o-alignment aligned-rep-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime alignment mask \\\n",
    "  --i-alignment aligned-rep-seqs.qza \\\n",
    "  --o-masked-alignment masked-aligned-rep-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime phylogeny fasttree \\\n",
    "  --i-alignment masked-aligned-rep-seqs.qza \\\n",
    "  --o-tree unrooted-tree.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime phylogeny midpoint-root \\\n",
    "  --i-tree unrooted-tree.qza \\\n",
    "  --o-rooted-tree rooted-tree.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Core Diversity Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny rooted-tree.qza \\\n",
    "  --i-table biom_id2.qza \\\n",
    "  --p-sampling-depth 4166 \\\n",
    "  --m-metadata-file UC_Severity_1_MF_Idswap6.21.18.txt \\\n",
    "  --output-dir core-metrics-results_idswap2_newmetadata_allsamples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Alpha diversity analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity core-metrics-results_idswap2/faith_pd_vector.qza \\\n",
    "  --m-metadata-file UC_Severity_1_MF_Idswap6.21.18.txt \\\n",
    "  --o-visualization core-metrics-results_idswap2_newmetadata/faith-pd-group-significance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity core-metrics-results_idswap2/evenness_vector.qza \\\n",
    "  --m-metadata-file UC_Severity_1_MF_Idswap6.21.18.txt \\\n",
    "  --o-visualization core-metrics-results_idswap2_newmetadata/evenness-group-significance.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Taxanomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime feature-classifier classify-sklearn \\\n",
    "  --i-classifier gg-13-8-99-515-806-nb-classifier.qza \\\n",
    "  --i-reads sequences.qza \\\n",
    "  --o-classification taxonomy.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime metadata tabulate \\\n",
    "  --m-input-file taxonomy.qza \\\n",
    "  --o-visualization taxonomy.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime taxa barplot \\\n",
    "  --i-table biom_id2.qza \\\n",
    "  --i-taxonomy taxonomy.qza \\\n",
    "  --m-metadata-file UC_Severity_1_MF_Idswap6.21.18.txt \\\n",
    "  --o-visualization taxa-bar-plots_idswap2.qzv"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
